---
title: "Data Files"
author: <a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a>
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    toc_depth: 2
---
<br>
<br>

```{r setup, message=FALSE, warning=FALSE, echo=FALSE}
library(DT)
```

## Antibiotic Resistance Genes {.tabset .tabset-pills}
### ARG Database
```{r args, message=FALSE, warning=FALSE, echo=FALSE}
args <- data.table::fread('../data/ARG_database.fa', header = FALSE, col.names = "Target ARGs")
datatable(args)
```
<a id="raw-url" href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/ARG_database.fa" download target="_blank">
  Download
</a>
<br>
<br>

### Forward Primers
```{r fprimers, message=FALSE, warning=FALSE, echo=FALSE}
fprimers <- data.table::fread('../data/forward_primers', header = FALSE, col.names = "Forward Primers")
datatable(fprimers)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/forward_primers" download target="_blank">
  Download
</a>
<br>
<br>

### Reverse Primers
```{r rprimers, message=FALSE, warning=FALSE, echo=FALSE}
rprimers <- data.table::fread('../data/reverse_primers', header = FALSE, col.names = "Reverse Primers")
datatable(rprimers)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/reverse_primers" download target="_blank">
  Download
</a>
<br>
<br>

### Target ARG Classifications
```{r classifications, message=FALSE, warning=FALSE, echo=FALSE}
classifications <- unique(data.table::fread('../data/target_ARG_classifications.tsv', header = TRUE))
datatable(classifications)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/target_ARG_classifications.tsv" download target="_blank">
  Download
</a>
<br>
<br>


## Mock-Community {.tabset .tabset-pills}
### Strain Info
```{r strains, message=FALSE, warning=FALSE, echo=FALSE}
strains <- data.table::fread('../data/mock_strain_infor.csv', drop = 1)
datatable(strains, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '150px',
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 23 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 23) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/mock_strain_infor.csv" download target="_blank">
  Download
</a>
<br>
<br>

### Genomes
```{r genomes, message=FALSE, warning=FALSE, echo=FALSE}
mock_genomes <- data.table::fread('../data/mock_genomes.fa', header = FALSE, col.names = "Mock Community Members")
mock_genome_names <- unname(unlist(ssBLAST:::fasta_seq_names("../data/mock_genomes.fa")))
mock_genomes
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/mock_genomes.fa" download target="_blank">
  Download
</a>
<br>
<br>


## Sample Info {.tabset .tabset-pills}
### Metadata
```{r metadata, message=FALSE, warning=FALSE, echo=FALSE}
metadata <- data.table::fread('../data/sample_metadata.csv')
metadata[[1]] <- gsub('Q3_', '', metadata[[1]])
metadata[[1]] <- gsub('_', '-', metadata[[1]])
data.table::set(metadata, j = 'Matrix', value = factor(metadata[[2]], levels = unique(metadata[[2]])))
datatable(metadata, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '1200px'
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/sample_metadata.csv" download target="_blank">
  Download
</a>
<br>
<br>


### ARG Counts {.tabset .tabset-pills}

#### Mock-Community
```{r mock_counts, message=FALSE, warning=FALSE, echo=FALSE}
read_counts <- readRDS('../data/read_counts_from_primers.RDS')
datatable(read_counts[ARG != "16S",c(1,2,3,grep('Mock', colnames(read_counts))), with=FALSE], 
                        options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(1,4:ncol(read_counts[ARG != "16S",c(1,2,3,grep('Mock', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>

#### Soil Column I
```{r sci_counts, message=FALSE, warning=FALSE, echo=FALSE}
datatable(read_counts[ARG != "16S",c(1,2,3,grep('SCI-', colnames(read_counts))), with=FALSE], 
                      options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[ARG != "16S",c(1,2,3,grep('SCI-', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>
  
#### Soil Column II
```{r scii_counts, message=FALSE, warning=FALSE, echo=FALSE}
datatable(read_counts[ARG != "16S",c(1,2,3,grep('SCII', colnames(read_counts))), with=FALSE], 
            options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[ARG != "16S",c(1,2,3,grep('SCII', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>

#### Fecal
```{r fecal_counts, message=FALSE, warning=FALSE, echo=FALSE}
datatable(read_counts[ARG != "16S",c(1,2,3,grep('X1P', colnames(read_counts))), with=FALSE], 
          options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[ARG != "16S",c(1,2,3,grep('X1P', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>

```{r curate_read_counts, echo=FALSE, warning=FALSE}
read_counts <- read_counts[ARG != "16S"]
low_samples <- names(which(colSums(read_counts[,-c(1:3)], na.rm = T) < 5000))
read_counts[, (low_samples) := NULL]
```

### Sequencing Data
<br>
<br>

FASTQ files for all data analyze for this study can be found at ...

<br>
<br>


## Soil Column Data

### Soil Column I
```{r sci, message=FALSE, warning=FALSE, echo=FALSE}
soil_column_I_ARGS <- readRDS('../data/soil_column_I_ARGs.RDS')
```

```{r curate_sci, echo=FALSE, warning=FALSE}
SC_I <- phylosmith::melt_phyloseq(soil_column_I_ARGS)
SC_I <- SC_I[Sample %in% SC_I[,sum(Abundance), by = c('Sample')]$Sample[SC_I[,sum(Abundance), by = c('Sample')]$V1 > 1000]]
# SC_I <- SC_I[Matrix %in% c('Manure', 'Soil') & Treatment == 'Control']
SC_I <- SC_I[, sum(Abundance), by = c('Sample', 'Gene', 'Treatment', 'Matrix', 'Day')]
data.table::set(SC_I, j = 'Gene', value = tolower((SC_I[['Gene']])))
SC_I[, ARG_Family := tolower(substr(SC_I[['Gene']],1,3))]
data.table::set(SC_I, j = 'Matrix', value = tolower(SC_I[['Matrix']]))
data.table::set(SC_I, which(SC_I[['Matrix']]=='soil' & SC_I[['Day']]==0), j='Matrix', value='Soil A')
data.table::set(SC_I, which(SC_I[['Matrix']]=='soil' & SC_I[['Day']]!=0), j='Matrix', value='Manure-Soil A')
SC_I$Sample <- factor(SC_I$Sample, levels = unique(SC_I$Sample[order(as.numeric(SC_I$Day))]))
data.table::setnames(SC_I, 'Gene', 'ARG')
```
