---
title: "Data Files"
author: <a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a>
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
    toc_depth: 2
editor_options: 
  chunk_output_type: console
---
<br>
<br>

```{r setup, echo=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```

## Antibiotic Resistance Genes {.tabset .tabset-pills}
### ARG Database
```{r args}
args <- data.table::fread('../data/ARGs/ARG_database.fa', header = FALSE, col.names = "Target ARGs")
```

```{r args_out, echo=FALSE}
datatable(args)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/ARGs/ARG_database.fa" download target="_blank">
  Download
</a>
<br>
<br>

### Primer Details
```{r primer_info}
primer_info <- data.table::fread('../data/primers/DARTE-QM_primer_design.csv', header = TRUE)
```

```{r primer_info_out, echo=FALSE}
datatable(primer_info, options = list(
            columnDefs = list(
              list(
                targets = 10,
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 12 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 12) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/DARTE-QM_primer_design.csv" download target="_blank">
  Download
</a>
<br>
<br>

### Forward Primers
```{r fprimers}
fprimers <- data.table::fread('../data/primers/forward_primers', header = FALSE, col.names = "Forward Primers")
```

```{r fprimers_out, echo=FALSE}
datatable(fprimers)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/forward_primers" download target="_blank">
  Download
</a>
<br>
<br>

### Reverse Primers
```{r rprimers}
rprimers <- data.table::fread('../data/primers/reverse_primers', header = FALSE, col.names = "Reverse Primers")
```

```{r rprimers_out, echo=FALSE}
datatable(rprimers)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/reverse_primers" download target="_blank">
  Download
</a>
<br>
<br>

### Target ARG Classifications
```{r classifications}
classifications <- unique(data.table::fread('../data/ARGs/target_ARG_classifications.tsv', header = TRUE))
```

```{r classifications_out, echo=FALSE}
datatable(classifications)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/ARGs/target_ARG_classifications.tsv" download target="_blank">
  Download
</a>
<br>
<br>


## Mock-Community {.tabset .tabset-pills}
### Strain Info
```{r strains}
strains <- data.table::fread('../data/mock/mock_strain_infor.csv', drop = 1)
```

```{r strains_out, echo=FALSE}
datatable(strains, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '150px',
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 23 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 23) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/mock/mock_strain_infor.csv" download target="_blank">
  Download
</a>
<br>
<br>

### Genomes
```{r genomes}
mock_genomes <- data.table::fread('../data/mock/mock_genomes_and_plasmids.fa', header = FALSE, col.names = "Mock Community Members")
mock_genome_names <- unname(unlist(ssBLAST:::fasta_seq_names("../data/mock/mock_genomes_and_plasmids.fa")))
```

```{r genomes_out, echo=FALSE}
mock_genomes
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/mock/mock_genomes_and_plasmids.fa" download target="_blank">
  Download
</a>
<br>
<br>


## Sample Info {.tabset .tabset-pills}
### Metadata
```{r metadata}
metadata <- data.table::fread('../data/samples/sample_metadata.csv')
metadata[['ID']] <- gsub('Q3_', '', metadata[['ID']])
metadata[['ID']] <- gsub('_', '-', metadata[['ID']])
data.table::set(metadata, j = 'Sample', value = factor(metadata[['Sample']], levels = unique(metadata[['Sample']])))
data.table::set(metadata, j = 'Matrix', value = factor(metadata[['Matrix']], levels = unique(metadata[['Matrix']])))
```

```{r metadata_out, echo=FALSE}
datatable(metadata, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '1200px'
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/samples/sample_metadata.csv" download target="_blank">
  Download
</a>
<br>
<br>


### ARG Counts {.tabset .tabset-pills}

#### Mock-Community
```{r mock_counts}
read_counts <- readRDS('../data/primers/read_counts_from_primers.RDS')
```

```{r mock_counts_out, echo=FALSE}
datatable(read_counts[Gene != "16S",c(1,2,3,grep('Mock', colnames(read_counts))), with=FALSE], 
                        options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(1,4:ncol(read_counts[Gene != "16S",c(1,2,3,grep('Mock', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>

#### Soil Column I
```{r sci_counts}
sc_i <- read_counts[Gene != "16S",c(1,2,3,grep('SCI-', colnames(read_counts))), with=FALSE]
```

```{r sci_counts_out, echo=FALSE}
datatable(sc_i, options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[Gene != "16S",c(1,2,3,grep('SCI-', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>
  
#### Soil Column II
```{r scii_counts}
sc_ii <- read_counts[Gene != "16S",c(1,2,3,grep('SCII', colnames(read_counts))), with=FALSE]
```

```{r scii_counts_out, echo=FALSE}
datatable(sc_ii, options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[Gene != "16S",c(1,2,3,grep('SCII', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primers/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>
â™«
#### Fecal
```{r fecal_counts}
fecal <- read_counts[Gene != "16S",c(1,2,3,grep('X1P', colnames(read_counts))), with=FALSE]
```

```{r fecal_counts_out, echo=FALSE}
datatable(fecal, 
          options = list(
            autoWidth = TRUE,
            columnDefs = list(
              list(width = '100px', targets = c(4:ncol(read_counts[Gene != "16S",c(1,2,3,grep('X1P', colnames(read_counts))), with=FALSE]))),
              list(
                targets = c(1,2),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 6 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 6) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = c(3),
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 4 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 4) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<a href="https://github.com/schuyler-smith/DARTE-QM/raw/master/data/primners/read_counts_from_primers.RDS" download target="_blank">
  Download
</a>
<br>
<br>

```{r curate_read_counts, echo=FALSE, warning=FALSE}
read_counts <- read_counts[Gene != "16S"]
low_samples <- names(which(colSums(read_counts[,-c(1:4)], na.rm = T) < 5000))
read_counts[, (low_samples) := NULL]
```

### Sequencing Data
<br>
<br>

FASTQ files for all data analyze for this study can be found at ...

<br>
<br>


## Soil Column Data

### Soil Column I
```{r sci_metag}
soil_column_I_ARGS <- readRDS('../data/samples/soil_column_I_ARGs.RDS')

SC_I <- phylosmith::melt_phyloseq(soil_column_I_ARGS)
SC_I <- SC_I[Sample %in% SC_I[,sum(Abundance), by = c('Sample')]$Sample[SC_I[,sum(Abundance), by = c('Sample')]$V1 > 1000]]
SC_I <- SC_I[, sum(Abundance), by = c('Sample', 'Gene', 'Treatment', 'Matrix', 'Day')]
data.table::set(SC_I, j = 'Gene', value = tolower((SC_I[['Gene']])))
SC_I[, ARG_Family := tolower(substr(SC_I[['Gene']],1,3))]
data.table::set(SC_I, j = 'Matrix', value = tolower(SC_I[['Matrix']]))
data.table::set(SC_I, which(SC_I[['Matrix']]=='soil' & SC_I[['Day']]==0), j='Matrix', value='Soil A')
data.table::set(SC_I, which(SC_I[['Matrix']]=='soil' & SC_I[['Day']]!=0), j='Matrix', value='Manure-Soil A')
SC_I$Sample <- factor(SC_I$Sample, levels = unique(SC_I$Sample[order(as.numeric(SC_I$Day))]))
data.table::setnames(SC_I, 'Gene', 'ARG')
```


```{r sci_metag_out, echo=FALSE, warning=FALSE, echo=FALSE}
SC_I
```