---
title: "Mock-Community Analysis"
author: <a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a>
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
editor_options: 
  chunk_output_type: console
---
<br>
<br>


```{r setup, message=FALSE, warning=FALSE, echo=FALSE}
library(data.table)
library(ggplot2)
library(data.table)
```

```{r create_mock_counts, message=FALSE, warning=FALSE, echo=FALSE, eval=FALSE}
rrna <- c('HM798937.1.1383','LC124582.1.1424','GQ927630.1.918','FJ680748.1.1374',
          'FJ455902.1.1529','KY355700.1.900','KP308750.18552.19877','MH337785.1',
          'AY779826.1.900','FR685529.1.1448','FJ365469.1.1342','JQ278775.1.1454',
          'MH938429.1','FJ363919.1.1352','FJ516926.1.1472','JN521105.1.1285',
          'KP090991.1.1467','FJ716960.1.1550','MH938447.1','KT381382.1.900',
          'KY898052.1','KY898059.1','FPLP01012084.16.1433','FJ478940.1.1496',
          'b0463','SL1344_RS02425')

mock_blast <- data.table::fread('../../data/mock/reverse_primers.blast')

mock_blast <- mock_blast[!(V1 %in% rrna)]

mock_blast <- mock_blast[V4 > 20]
mock_blast <- mock_blast[V3 > 95]
setorder(mock_blast, 'V2', 'V9')

strains[, Abundance := Genome_copies_per_microgram * micrograms]

mock_genome <- vector()
for(genome in mock_blast$V2){
	genome <- mock_genome_names[sapply(strsplit(mock_genome_names, ' '), FUN=function(x){any(x %in% genome)})]
	genome <- strains[apply(strains,1,FUN=function(x){any(x %in% unlist(strsplit(genome, " ")))})]$Strain
	mock_genome <- c(mock_genome, genome)
} 

mock_genomes[, ARG := classifications[[2]][match(mock_genomes[['V1']], classifications[[3]])]]
mock_genomes[, ARG_Family := classifications[[1]][match(mock_genomes[['V1']], classifications[[3]])]]
mock_genomes$V4[-c(1,2,39,41,42,44,45,57,82,83)]
a <- mock_genomes[,c('V2', 'V1', 'V9', 'ARG', 'ARG_Family')]

setnames(a, c('V2','V1'), c('Microbe', 'Primer'))

a[, abundance := mock_info[unlist(sapply(a[[1]], grep, mock_info$Strain))]$Abundance]
a$abundance[1:3] <- a$abundance[1:3]/3
a$abundance[39:40] <- a$abundance[39:40]/2
a$abundance[41:43] <- a$abundance[41:43]/2
a$abundance[44:46] <- a$abundance[44:46]/3
a$abundance[57:58] <- a$abundance[57:58]/2
a$abundance[82:84] <- a$abundance[82:84]/3

a <- a[, .(count = sum(abundance)), by = c( 'ARG_Family', "ARG", 'Primer', "Microbe")]
a[, count := (count/sum(count))]
# a
a[,Sample := 'Theoretical']
setcolorder(a, c("Sample", "Microbe", "ARG", "ARG_Family", 'Primer', "count"))
saveRDS(a, '../data/mock_ARG_counts.rds')
mock_ARGs <- readRDS('../data/mock_ARG_counts.rds')
set(mock_ARGs, j = "Primer", value = gsub('\\:', '_', mock_ARGs$Primer))
expected_mock <- mock_ARGs$Primer
mock_samples <- metadata[grep('Mock', metadata$`Sample Name`),][Analyzed == TRUE]$`Sample Name`


```


## Synthetic Oligonucleotide 'Spike'  {.tabset .tabset-pills}

```{r setup_spike, message=FALSE, warning=FALSE, echo=FALSE}
spike_counts <- read_counts[ARG == 'spike']
mock_samples <- metadata[grep('Mock', metadata$`Sample Name`),][Analyzed == TRUE]$`Sample Name`
```

```{r spike_counts, echo=FALSE, message=FALSE, warning=FALSE}
graph_data <- melt(spike_counts, id.vars = c('ARG_Family','ARG','Primer'))
graph_data <- graph_data[, floor(mean(value)), by = c('variable')]
graph_data <- graph_data[variable %in% mock_samples]
graph_data[['variable']] <- factor(graph_data[['variable']], levels = mock_samples)
setnames(graph_data, colnames(graph_data), c("Sample", "Spike"))
graph_data <- cbind(graph_data, Spike_Level = as.numeric(gsub('-','.',gsub('Mock-','',gsub('spike-.?','', graph_data$Sample)))))
setkey(graph_data, 'Spike_Level')
graph_data[, Color := schuylR::create_palette(length(unique(graph_data[['Spike_Level']])))[as.factor(graph_data[['Spike_Level']])]]
spike_bars <-  ggplot(graph_data,
                      aes_string(x = 'Sample', y = 'Spike')) + 
  geom_bar(stat = "identity", size = 0.45, alpha = 1, position = "dodge",
           fill = graph_data$Color) + 
  theme_light() +
  theme(
    axis.line.x = element_line(
      colour = 'black',
      size = 1,
      linetype = 'solid'
    ),
    axis.line.y = element_line(
      colour = 'black',
      size = 1,
      linetype = 'solid'
    ),
    axis.text.x = element_text(
      size = 10,
      vjust = 1,
      hjust = 1,
      angle = 30
    ),
    axis.title.x = element_text(size = 12, face = "bold"),
    axis.title.y = element_text(size = 12, face = "bold"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12, face = "bold"),
    legend.background = element_rect(fill = (alpha = 0)),
    legend.key.size = unit(4, "mm"),
    legend.spacing.x = unit(0.005, 'npc'),
    strip.text.x = element_text(size = 10, face = 'bold', color = 'black'),
    strip.background = element_rect(colour = 'black', size = 1.4, fill = 'white')
  ) +
  scale_y_continuous(expand = expansion(mult = c(0.0025, 0.01))) +
  ylab("Read Count of Spike Sequence") + 
  xlab("")
```

```{r spike_counts_lines, echo=FALSE, message=FALSE, warning=FALSE}
spike_points <-  ggplot(graph_data,
                        aes_string(x = 'Spike_Level', y = 'Spike')) + 
  geom_smooth(formula = y~x, method = "lm", se = FALSE, aes(x = Spike_Level, y = Spike), color = 'black' ) + 
  geom_point(stat = "identity", size = 4, alpha = 1, position = position_dodge(width = 0),
             color = graph_data$Color) + 
  theme_bw() + theme(axis.text.x = element_text(
    hjust = 1, size = 10), axis.text.y = element_text(hjust = 0.95, 
                                                      size = 10), axis.title.x = element_text(size = 12, face = "bold"), 
    axis.title.y = element_text(size = 12, face = "bold"), 
    axis.ticks.x = element_blank(), legend.title = element_text(size = 10, 
                                                                face = "bold"), legend.text = element_text(size = 8), 
    legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4, 
                                                                  "mm"), legend.background = element_rect(fill = (alpha = 0)), 
    panel.background = element_rect(color = "black", size = 1.5, 
                                    fill = "white"), panel.spacing = unit(0.015, "npc"), 
    strip.text.x = element_text(size = 10, face = "bold", 
                                color = "black"), strip.background = element_rect(colour = "black", 
                                                                                  size = 1.4, fill = "white")) + 
  labs(x = "Spike Level", y = "Spike Sequences Detected at >= 98% Identity")
```

### Linear Regression
```{r spike_counts_line_reg, fig.width=6, fig.height=4, echo=FALSE, warning=FALSE}
spike_points
```
<br>
<br>

### Bar Chart
```{r spike_counts_bars, fig.width=6, fig.height=4, echo=FALSE, warning=FALSE}
spike_bars
```
<br>
<br>

### Table
```{r spike_counts_table, echo=FALSE, warning=FALSE}
datatable(graph_data[,-c('Color')])
```
<br>
<br>