---
title: "Primer Analysis"
author: <a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a>
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
    toc_depth: 2
editor_options: 
  chunk_output_type: console
---
<br>
<br>


## Primer Success Tables {.tabset .tabset-pills}
```{r create_succes_file, eval=FALSE}
BLAST_file_path = "../blast/"

if (length(BLAST_file_path) == 1 && dir.exists(BLAST_file_path)) {
  BLAST_file_path <- gsub("/$", "", BLAST_file_path)
  BLAST_file <- dir(BLAST_file_path, full.names = TRUE)
} else {
  BLAST_file <- BLAST_file_path
}
BLAST_file <- sort(normalizePath(BLAST_file))

primer_success <- data.table::data.table(sample = character(),
                                         primer = character(),
                                         n_reads = numeric(),
                                         artifacts = numeric(),
                                         analyzed = numeric(),
                                         true_pos = numeric(),
                                         false_pos = numeric(),
                                         false_neg = numeric(),
                                         true_neg = numeric())

for(file in BLAST_file){
  sample_name <- gsub('Q3-','',basename(file))
  files <- dir(file, full.names = TRUE)
  files <- sort(normalizePath(files))
  
  unknown <- files[grep('unknown', files)]
  unknown <- data.table::fread(unknown, header = FALSE, sep = '\t',
                             col.names = c('query', 'reference', 'perc_id', 'length', 
                                           'mismatch', 'gap', 'qstart', 'qend', 
                                           'rstart', 'rend', 'e', 'bit'))
  unknown <- unknown[length >= 90]
  setorder(unknown, 'query', 'perc_id')
  unknown <- merge(classifications, unknown, by.x = "Primer", by.y = "reference")
  unknown <- unique(unknown, by=c("query", "Primer"))
  unknown <- unknown[unknown[, .I[length == max(length, na.rm=T)], by = c('query', 'Primer')][[3]]]
  unknown <- unknown[unknown[, .I[perc_id == max(perc_id, na.rm=T)], by = c('query', 'Primer')][[3]]]
  unknown <- unknown[, .(count = .N), by = Primer]
  
  for(primer_file in files){
    if(file.info(primer_file)$size > 0){
      blast <- data.table::fread(primer_file, header = FALSE, sep = '\t',
                                 col.names = c('query', 'reference', 'perc_id', 'length', 
                                               'mismatch', 'gap', 'qstart', 'qend', 
                                               'rstart', 'rend', 'e', 'bit'))
      setorder(blast, 'query', 'perc_id')
      blast <- blast[perc_id > 95]
      blast <- merge(classifications, blast, by.x = "Primer", by.y = "reference")
      n_reads <- length(unique(blast[['query']]))
      blast <- blast[length >= 90]
      analyzed <- length(unique(blast[['query']]))
      n_artifacts <- n_reads - analyzed
      primer_target <- gsub('____.?', '', basename(sapply(strsplit(primer_file, "\\."), "[[", 1)))
      false_neg <- unknown[Primer == primer_target][[2]]
      successes <- 0
      if(nrow(blast) > 0){
        blast <- blast[blast[, .I[length == max(length, na.rm=T)], by = c('query', 'Primer')][[3]]]
        blast <- unique(blast, by=c("query", "Primer"))
        blast <- blast[, .(count = .N), by = Primer]
        successes <- blast[Primer == primer_target][[2]]
      }
      if(length(successes) < 1) successes <- 0
      if(length(false_neg) < 1) false_neg <- 0
      primer_success <- rbind(primer_success,
                              data.table::data.table(
                                sample = sample_name,
                                primer = primer_target,
                               n_reads = n_reads,
                               artifacts = n_artifacts,
                               analyzed = analyzed,
                               true_pos = successes,
                               false_pos = (analyzed - successes),
                               true_neg = 0,
                               false_neg = false_neg))
    }
  }
  set(primer_success, primer_success[, .I[sample == sample_name]], 'true_neg', sum())
  print(sample_name)
}
primer_success[,true_neg := sum(true_pos)-true_pos, by = sample]
primer_success <- merge(classifications, primer_success, by.x = 'Primer', by.y = 'primer')
primer_success <- primer_success[n_reads != 'NA']
primer_success <- primer_success[ARG_Family != '16S']
saveRDS(primer_success, '../data/primers/primer_success.RDS')
data.table::set(primer_success, which(primer_success[["success_rate"]] == 0), "success_rate", NA)
primer_success[, experiment := sapply(strsplit(sample, "-"), "[[", 1)]
read_count <- data.table::dcast(primer_success, Primer ~ sample, value.var=c("true_pos"), fun = sum)
read_count <- read_count[c(which(rowSums(read_count[,-1]) > 0)),]
read_count <- merge(classifications, read_count, by = 'Primer')
saveRDS(read_count, '../data/primers/read_counts_from_primers.RDS')
```

### Primers by Samples
```{r readin_success, echo=FALSE}
primer_success <- readRDS('../data/primers/primer_success.RDS')
```

```{r success_table, echo=FALSE, cache=TRUE}
primer_success
```
<a href="../data/primer_success.RDS" download>
  Download
</a>
<br>
<br>


### Overall for Primers
```{r primers}
samples <- primer_success[, sum(true_pos, na.rm = T), by = sample][V1 > 1000]$sample
primer_success <- primer_success[sample %in% samples]
primers <- primer_success[, lapply(.SD, sum, na.rm=TRUE), by=Primer, .SDcols=c("n_reads", "artifacts", "analyzed", "true_pos", "false_pos") ] 
setorder(primers, -n_reads)
```

```{r primers_table, echo=FALSE}
datatable(primers, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '260px',
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 20 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 20) + '...</span>' : data;",
                  "}")
              ),
              list(
                targets = 3,
                width = '100px',
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 12 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 12) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<br>
<br>

### Overall for Samples
```{r samples}
sample_success <- primer_success[, lapply(.SD, sum, na.rm=TRUE), by=sample, .SDcols=c("n_reads", "artifacts", "analyzed", "true_pos", "false_pos") ] 
setorder(sample_success, sample)
```

```{r samples_table, echo=FALSE}
datatable(sample_success, options = list(
            columnDefs = list(
              list(
                targets = 1,
                width = '260px',
                render = JS(
                  "function(data, type, row, meta) {",
                  "return type === 'display' && data.length > 20 ?",
                  "'<span title=\"' + data + '\">' + data.substr(0, 20) + '...</span>' : data;",
                  "}")
              )
            )
          )
)
```
<br>
<br>

### True-Positive Reads by Samples
```{r sample_t_pos}
data.table::set(primer_success, which(primer_success[["success_rate"]] == 0), "success_rate", NA)
primer_success[, experiment := sapply(strsplit(sample, "-"), "[[", 1)]
read_count <- data.table::dcast(primer_success, Primer ~ sample, value.var=c("true_pos"), fun = sum)
read_count <- read_count[c(which(rowSums(read_count[,-1]) > 0)),]
read_count <- merge(classifications, read_count, by = 'Primer')
```

```{r t_pos_counts, echo=FALSE}
read_count
```
<br>
<br>


## Success Metrics {.tabset .tabset-pills}
```{r primer_table}
primer_success[, experiment := NULL]
setorder(primer_success, -n_reads)
```
<br>

```{r metrics, class.source = 'fold-show'}
primer_success[, sensitivity := round(true_pos/(true_pos + false_neg)*100, 1)]
primer_success[, specificity := round(true_neg/(true_neg + false_pos)*100, 1)]
primer_success[, accuracy := round(true_pos/(true_pos + false_neg)*100, 1)]
```
<br>

```{r success_metrics, echo=FALSE}
primer_success
```
<br>
<br>

### Mean Sensitivity

Mean sensitivity of each primer, weighted by abundance per sample, was found to be **`r round(weighted.mean(primer_success$sensitivity, primer_success$analyzed, na.rm = TRUE), 2)`**
<br>
<br>

### Mean Specificity

Mean specificity of each primer, weighted by abundance per sample, was found to be **`r round(weighted.mean(primer_success$specificity, primer_success$analyzed, na.rm = TRUE), 2)`**
<br>
<br>

### Mean Accuracy
Mean accuracy of each primer, weighted by abundance per sample, was found to be **`r round(weighted.mean(primer_success$accuracy, primer_success$analyzed, na.rm = TRUE), 2)`**
<br>
<br>


## Primer Performance Graphs {.tabset .tabset-pills}
```{r overall_primer}
primers <- primer_success[, lapply(.SD, sum, na.rm=TRUE), by=Primer, .SDcols=c("n_reads", "artifacts", "analyzed", "true_pos", "false_pos", "false_neg", "true_neg") ] 
primers[, percent_artifacts := artifacts/n_reads*100]
primers[, percent_tp := true_pos/n_reads*100]
primers[, percent_fp := false_pos/n_reads*100]
primers[, percent_fn := false_neg/n_reads*100]
```

### All
```{r primer_artifact_graph, fig.width=12, fig.height=7, echo=FALSE, cache=TRUE}
setorder(primers, -percent_artifacts)
set(primers, j = 'Primer', value = factor(primers$Primer, levels = primers$Primer))
colors <- schuylR::create_palette(5)

ggplot(primers, aes_string(x="Primer", y="percent_artifacts")) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(fill = "Artifact"), alpha = 0.7) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_tp, fill = "True-Positive"), alpha = 0.7 ) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_fp, fill = "False-Positive"), alpha = 0.7 ) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_fn, fill = "False Negative"), alpha = 0.7 ) + 
  theme_bw() + 
  theme(
    legend.position = c(.5, .85),
    axis.text.x = element_text(size = 4, angle = 30, hjust = 0.95)
  ) + 
  scale_y_discrete(expand=expansion(mult=0, add=1), limits = seq(0,100,10)) + 
  ylab("Percentage of Reads") + 
  scale_fill_manual(values = colors[c(2,4,5,3)], name = "Read Classification")
```
<br>

### Artifact
```{r artifact_graph, fig.width=12, fig.height=7, echo=FALSE, cache=TRUE}
ggplot(primers, aes_string(x="Primer", y="percent_artifacts")) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(fill="Artifact")) + 
  theme_bw() + 
  theme(
    legend.position = c(.5, .90),
    axis.text.x = element_text(size = 4, angle = 30, hjust = 0.95)
  ) + 
  scale_y_discrete(expand=expansion(mult=0, add=1), limits = seq(0,100,10)) + 
  ylab("Percentage of Reads") + 
  scale_fill_manual(values = colors[c(2)], name = "Read Classification")
```
<br>

### True-Positive
```{r tp_graph, fig.width=12, fig.height=7, echo=FALSE, cache=TRUE}
ggplot(primers, aes_string(x="Primer", y="percent_artifacts")) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_tp, fill="True-Positive") ) + 
  theme_bw() + 
  theme(
    legend.position = c(.5, .90),
    axis.text.x = element_text(size = 4, angle = 30, hjust = 0.95)
  ) + 
  scale_y_discrete(expand=expansion(mult=0, add=1), limits = seq(0,100,10)) + 
  ylab("Percentage of Reads") + 
  scale_fill_manual(values = colors[c(3)], name = "Read Classification")
```
<br>

### False-Positive
```{r fp_graph, fig.width=12, fig.height=7, echo=FALSE, cache=TRUE}
ggplot(primers, aes_string(x="Primer", y="percent_artifacts")) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_fp, fill="False-Positive") ) + 
  theme_bw() + 
  theme(
    legend.position = c(.5, .90),
    axis.text.x = element_text(size = 4, angle = 30, hjust = 0.95)
  ) + 
  scale_y_discrete(expand=expansion(mult=0, add=1), limits = seq(0,100,10)) + 
  ylab("Percentage of Reads") + 
  scale_fill_manual(values = colors[c(4)], name = "Read Classification")
```
<br>

### False-Negative
```{r fn_graph, fig.width=12, fig.height=7, echo=FALSE, cache=TRUE}
ggplot(primers, aes_string(x="Primer", y="percent_artifacts")) + 
  geom_bar(stat="identity", position="stack", 
           size=0.12, width=0.95, aes(y = percent_fn, fill="False-Negative") ) + 
  theme_bw() + 
  theme(
    legend.position = c(.5, .90),
    axis.text.x = element_text(size = 4, angle = 30, hjust = 0.95)
  ) + 
  scale_y_discrete(expand=expansion(mult=0, add=1), limits = seq(0,100,10)) + 
  ylab("Percentage of Reads") + 
  scale_fill_manual(values = colors[c(5)], name = "Read Classification")
```
<br>

## Artifact Reads

### Examples of Artifact Reads
```{r example_artifacts, echo=FALSE}
fread('../data/primers/example_artifacts.fa', header = FALSE, col.names = "SCI-Soil-D59-A R1 Reads")
```

<br>
<br>

### Artifacts by Primer Count
```{r by_reads, echo=FALSE}
by_true_pos <- data.frame(sensitivity = 0, number_reads = seq(0, 1000,10), type = "True-Positive Reads")
for(i in seq_along(by_true_pos$number_reads)){
  by_true_pos[i,1] <- mean(primer_success[true_pos > by_true_pos$number_reads[i]]$sens)
}
```
<br>

```{r by_read_graph, fig.width=6, fig.height=4, echo=FALSE, cache=TRUE}
ggplot(rbind(by_true_pos), aes_string(x = "number_reads", y = "sensitivity", color="type")) +
 geom_line(size = 1.6, show.legend = F) + 
  scale_colour_manual(values = c("#56B4E9")) + theme_bw() + 
  theme(axis.text.x = element_text(size = 10), 
        axis.text.y = element_text(size = 10),  
        axis.title.x = element_text(size = 12, face = "bold"), 
        axis.title.y = element_text(size = 12, face = "bold"), 
        axis.ticks.x = element_blank(), 
        panel.background = element_rect(color = "black", size = 1.5, fill = "white"),
        panel.grid.major = element_line(size=0.1, color = 'grey50')) +
  scale_x_continuous(breaks = seq(0,1000,100),
                     expand = expansion(mult = 0.01, 
                                        add = c(0.5, 0.5))) +
  # scale_y_continuous(breaks = seq(0,100,5),
  #                    expand = expansion(add = c(0.5, 0.5))) +
  labs(y = "Rate of Success \n (ratio of true-positive reads)",
       x = "Minimum Count per Primer",
       color = "")
```
<br>
<br>

### Correlation of Artifacts and Read Count
```{r artifacts, echo=FALSE}
samples <- primer_success[, lapply(.SD, sum, na.rm=TRUE), by=sample, .SDcols=c("n_reads", "artifacts") ] 
samples[,artifacts := artifacts/n_reads*100]
setorder(samples, sample)
```
<br>

The number of reads produced per sample and the percentage of those reads that were PCR-artifacts had a negative correlation with an R-squared value = **`r round(summary(lm(artifacts ~ n_reads, samples))$r.squared, 2)`**

```{r artifacts_graph, fig.width=7, fig.height=5, echo=FALSE, cache=TRUE}
  ggplot(merge(metadata, samples,by.x='ID', by.y='sample'), aes_string(x = "n_reads", y = "artifacts", color="Matrix")) +
   geom_point(size = 4, show.legend = T) + 
  geom_smooth(formula = y~x, method = "lm", se = FALSE, color = 'black' ) +
  scale_colour_manual(values = schuylR::create_palette(length(unique(metadata$Matrix)))) + theme_bw() + 
  theme(axis.text.x = element_text(size = 10), 
        axis.text.y = element_text(size = 10),  
        axis.title.x = element_text(size = 12, face = "bold"), 
        axis.title.y = element_text(size = 12, face = "bold"), 
        axis.ticks.x = element_blank(), 
        panel.background = element_rect(color = "black", size = 1.5, fill = "white"),
        panel.grid.major = element_line(size=0.1, color = 'grey50')) +
  labs(y = "Percentage of Artifact Reads in Sample",
       x = "Number of Reads in Sample",
       color = "")
```
<br>
<br>

