<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Schuyler D. Smith" />

<meta name="date" content="2021-04-12" />

<title>Comaprison of DARTE-QM to Metagenome</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.17/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DARTE-QM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li>
  <a href="sequencing.html">
    <span class="fas fa-dna"></span>
     
    Sequencing
  </a>
</li>
<li>
  <a href="primer_analysis.html">
    <span class="fas fa-barcode"></span>
     
    Primers
  </a>
</li>
<li>
  <a href="mock_analysis.html">
    <span class="fas fa-vial"></span>
     
    Mock
  </a>
</li>
<li>
  <a href="sample_analysis.html">
    <span class="fas fa-chart-pie"></span>
     
    Sample
  </a>
</li>
<li>
  <a href="metagenome_comparison.html">
    <span class="fas fa-greater-than-equal"></span>
     
    Metagenome
  </a>
</li>
<li>
  <a href="data.html">
    <span class="fas fa-database"></span>
     
    Data
  </a>
</li>
<li>
  <a href="packages_and_functions.html">
    <span class="fas fa-box-open"></span>
     
    Packages
  </a>
</li>
<li>
  <a href="https://github.com/schuyler-smith/DARTE-QM/">
    <span class="fas fa-code"></span>
     
    Source Code
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/schuyler-smith">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://germslab.org/">
    <span class="fas fa-flask"></span>
     
  </a>
</li>
<li>
  <a href="https://www.bcb.iastate.edu/people/schuyler-smith">
    <span class="fas fa-university"></span>
     
  </a>
</li>
<li>
  <a href="mailto:sdsmith@iastate.edu">
    <span class="fas fa-envelope fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comaprison of DARTE-QM to Metagenome</h1>
<h4 class="author"><a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a></h4>
<h4 class="date">2021-04-12</h4>

</div>


<p><br> <br></p>
<div id="all-args-found" class="section level2 tabset tabset-pills">
<h2>All ARGs Found</h2>
<div id="profiles" class="section level3">
<h3>Profiles</h3>
<pre class="r"><code>classification &lt;- &#39;ARG&#39;
treatment_name &lt;- &#39;Matrix&#39;

missing &lt;- unique(sort(as.character(SC_I$ARG)))
genes &lt;- as.character(SC_I$ARG)
for(i in seq_along(missing)){
  genes[genes %in% missing[i]] &lt;- c(&quot;AAC(2&#39;)-Ib&quot;, &quot;AAC(2&#39;)-Ie&quot;, &quot;AAC(3)-Ib&quot;, &quot;AAC(3)-IIc&quot;, &quot;AAC(3)-IVa&quot;, &quot;AAC(6&#39;)-Iae&quot;, &quot;AAC(6&#39;)-Iak&quot;, &quot;AAC(3)-Ib/AAC(6&#39;)-Ib&#39;&#39;&quot;, &quot;AAC(6&#39;)-Ib11&quot;, &quot;AAC(6&#39;)-Ie-APH(2&#39;&#39;)-Ia&quot;, &quot;AAC(6&#39;)-IIa&quot;, &quot;AAC(6&#39;)-Iid&quot;, &quot;AAC(6&#39;)-Ip&quot;, &quot;ANT(6)-Ia&quot;, &quot;aadA1&quot;, &quot;aadA11&quot;, &quot;aadA13&quot;, &quot;aadA14&quot;, &quot;aadA15&quot;, &quot;aadA16&quot;, &quot;aadA22&quot;, &quot;aadA24&quot;, &quot;aadA25&quot;, &quot;aadA3&quot;, &quot;aadA5&quot;, &quot;aadA6&quot;, &quot;aadA7&quot;, &quot;aadA8&quot;, &quot;aadA9&quot;, &quot;ACI-1&quot;, &quot;acrA&quot;, &quot;acrB&quot;, &quot;acrD&quot;, &quot;AcrF&quot;, &quot;AdeB&quot;, &quot;AdeF&quot;, &quot;AdeI&quot;, &quot;AdeJ&quot;, &quot;AIM-1&quot;, &quot;AmrA&quot;, &quot;AmrB&quot;, &quot;ANT(4&#39;)-Ib&quot;, &quot;ANT(6)-Ia&quot;, &quot;ANT(6)-Ib&quot;, &quot;ANT(9)-Ia&quot;, &quot;APH(2&#39;&#39;)-If&quot;, &quot;APH(2&#39;&#39;)-Ib&quot;, &quot;APH(2&#39;&#39;)-IIIa&quot;, &quot;APH(3&#39;&#39;)-Ib&quot;, &quot;APH(3&#39;)-Ia&quot;, &quot;APH(3&#39;)-IIa&quot;, &quot;APH(3&#39;)-IIIa&quot;, &quot;APH(3&#39;)-VI&quot;, &quot;APH(3&#39;)-VIa&quot;, &quot;APH(3&#39;)-XV&quot;, &quot;APH(4)-Ia&quot;, &quot;APH(6)-Ia&quot;, &quot;APH(6)-Ib&quot;, &quot;APH(6)-Ic&quot;, &quot;APH(6)-Id&quot;, &quot;AQU-1&quot;, &quot;arnA&quot;, &quot;ARR-1&quot;, &quot;AxyX&quot;, &quot;AxyY&quot;, &quot;BcI&quot;, &quot;BcII&quot;, &quot;bcr-1&quot;, &quot;BJP-1&quot;, &quot;blaA&quot;, &quot;Bla2&quot;, &quot;BlaB&quot;, &quot;blaF&quot;, &quot;blaL1&quot;, &quot;BUT-1&quot;, &quot;carA&quot;, &quot;carB-1&quot;, &quot;carB-4&quot;, &quot;carB-8&quot;, &quot;cat&quot;, &quot;cat-TC&quot;, &quot;cat&quot;, &quot;catB6&quot;, &quot;catB8&quot;, &quot;catQ&quot;, &quot;ceoA&quot;, &quot;ceoB&quot;, &quot;cfr(C)&quot;, &quot;cfxA3&quot;, &quot;cfxA6&quot;, &quot;clbA&quot;, &quot;cml&quot;, &quot;cmlV&quot;, &quot;cmx&quot;, &quot;cpxa&quot;, &quot;crp&quot;, &quot;CTX-M-63&quot;, &quot;CTX-M-75&quot;, &quot;dfrA1&quot;, &quot;dfrA20&quot;, &quot;dfrA21&quot;, &quot;dfrB3&quot;, &quot;dfrD&quot;, &quot;dfrG&quot;, &quot;efpA&quot;, &quot;efrA&quot;, &quot;ereB&quot;, &quot;Erm(33)&quot;, &quot;Erm(35)&quot;, &quot;Erm(36)&quot;, &quot;Erm(42)&quot;, &quot;Erm(47)&quot;, &quot;ErmA&quot;, &quot;ErmB&quot;, &quot;ErmC&quot;, &quot;ErmF&quot;, &quot;ErmG&quot;, &quot;ErmQ&quot;, &quot;ErmT&quot;, &quot;ErmX&quot;, &quot;ErmY&quot;, &quot;EXO-1&quot;, &quot;facT&quot;, &quot;farA&quot;, &quot;FEZ-1&quot;, &quot;floR&quot;, &quot;FosB&quot;, &quot;fusH&quot;, &quot;gadX&quot;, &quot;GES-23&quot;, &quot;gyrA&quot;, &quot;ileS&quot;, &quot;iri&quot;, &quot;JOHN-1&quot;, &quot;linG&quot;, &quot;lnuA&quot;, &quot;lnuB&quot;, &quot;lnuC&quot;, &quot;lnuD&quot;, &quot;lnuF&quot;, &quot;lnuG&quot;, &quot;LRA-1&quot;, &quot;LRA-13&quot;, &quot;LRA-19&quot;, &quot;LRA-3&quot;, &quot;LRA-8&quot;, &quot;LRA-9&quot;, &quot;lrfA&quot;, &quot;lsaB&quot;, &quot;lsaC&quot;, &quot;lsaE&quot;, &quot;macB&quot;, &quot;mdsB&quot;, &quot;mdsC&quot;, &quot;mdtA&quot;, &quot;mdtB&quot;, &quot;mdtC&quot;, &quot;mdtF&quot;, &quot;mdtG&quot;, &quot;mdtK&quot;, &quot;mdtN&quot;, &quot;mdtO&quot;, &quot;mefA&quot;, &quot;mefB&quot;, &quot;mefC&quot;, &quot;mel&quot;, &quot;MexA&quot;, &quot;MexB&quot;, &quot;MexC&quot;, &quot;MexD&quot;, &quot;MexE&quot;, &quot;MexF&quot;, &quot;MexI&quot;, &quot;MexK&quot;, &quot;MexL&quot;, &quot;MexM&quot;, &quot;MexN&quot;, &quot;MexQ&quot;, &quot;MexV&quot;, &quot;MexW&quot;, &quot;MexX&quot;, &quot;MexY&quot;, &quot;mgtA&quot;, &quot;MOX-9&quot;, &quot;mphB&quot;, &quot;mphD&quot;, &quot;mphG&quot;, &quot;mphL&quot;, &quot;mprF&quot;, &quot;msrA&quot;, &quot;msrE&quot;, &quot;mtrA&quot;, &quot;mtrD&quot;, &quot;mupA&quot;, &quot;MuxA&quot;, &quot;MuxB&quot;, &quot;MuxC&quot;, &quot;novA&quot;, &quot;npmA&quot;, &quot;OCH-2&quot;, &quot;OCH-6&quot;, &quot;OCH-7&quot;, &quot;OKP-B-20&quot;, &quot;oleB&quot;, &quot;oleC&quot;, &quot;oleD&quot;, &quot;opcM&quot;, &quot;OpmB&quot;, &quot;OpmE&quot;, &quot;OpmH&quot;, &quot;OprA&quot;, &quot;OprM&quot;, &quot;OprN&quot;, &quot;OprZ&quot;, &quot;optrA&quot;, &quot;oqxB&quot;, &quot;otrA&quot;, &quot;otrB&quot;, &quot;otrC&quot;, &quot;OXA-114a&quot;, &quot;OXA-209&quot;, &quot;OXA-22&quot;, &quot;OXA-258&quot;, &quot;OXA-29&quot;, &quot;OXA-347&quot;, &quot;OXA-368&quot;, &quot;OXA-415&quot;, &quot;OXA-60&quot;, &quot;OXA-74&quot;, &quot;patA&quot;, &quot;PDC-75&quot;, &quot;PmpM&quot;, &quot;pmrE&quot;, &quot;qacH&quot;, &quot;QepA4&quot;, &quot;QnrB12&quot;, &quot;RbpA&quot;, &quot;rgt1438&quot;, &quot;Rm3&quot;, &quot;rmtF&quot;, &quot;rphA&quot;, &quot;rphB&quot;, &quot;SAT-1&quot;, &quot;SAT-3&quot;, &quot;SAT-4&quot;, &quot;smeA&quot;, &quot;smeB&quot;, &quot;smeC&quot;, &quot;smeD&quot;, &quot;smeE&quot;, &quot;smeF&quot;, &quot;smeR&quot;, &quot;smeS&quot;, &quot;spd&quot;, &quot;SPG-1&quot;, &quot;srmB&quot;, &quot;str&quot;, &quot;sul1&quot;, &quot;sul2&quot;, &quot;TaeA&quot;, &quot;tap&quot;, &quot;tcmA&quot;, &quot;tcr3&quot;, &quot;TEM-220&quot;, &quot;tet(31)&quot;, &quot;tet(33)&quot;, &quot;tet(39)&quot;, &quot;tet(40)&quot;, &quot;tet(41)&quot;, &quot;tet(42)&quot;, &quot;tet(43)&quot;, &quot;tet(44)&quot;, &quot;tet(45)&quot;, &quot;tetA&quot;, &quot;tetC&quot;, &quot;tetD&quot;, &quot;tetG&quot;, &quot;tetH&quot;, &quot;tetJ&quot;, &quot;tetL&quot;, &quot;tetV&quot;, &quot;tet(W/N/W)&quot;, &quot;tetY&quot;, &quot;tetZ&quot;, &quot;tet(32)&quot;, &quot;tet(36)&quot;, &quot;tetA(46)&quot;, &quot;tetA(P)&quot;, &quot;tetB(P)&quot;, &quot;tetM&quot;, &quot;tetO&quot;, &quot;tetQ&quot;, &quot;tetS&quot;, &quot;tetT&quot;, &quot;tetW&quot;, &quot;tetX&quot;, &quot;tlrC&quot;, &quot;TolC&quot;, &quot;TriA&quot;, &quot;triC&quot;, &quot;VanHO&quot;, &quot;vanJ&quot;, &quot;vanO&quot;, &quot;vanRG&quot;, &quot;vanRI&quot;, &quot;VanRM&quot;, &quot;VanRO&quot;, &quot;VanSA&quot;, &quot;VanSF&quot;, &quot;VanSM&quot;, &quot;VanSO&quot;, &quot;VanTG&quot;, &quot;VanXO&quot;, &quot;VanXYG&quot;, &quot;vatE&quot;, &quot;vatH&quot;, &quot;vga(A)LC&quot;, &quot;vgaB&quot;, &quot;vgaE&quot;)[i]
}
set(SC_I, j = &#39;ARG&#39;, value = factor(genes, levels = unique(genes)))

graph_data &lt;- melt(read_counts, id.vars = c(&#39;ARG_Family&#39;, &#39;ARG&#39;, &#39;Primer&#39;), variable.name = &#39;Sample_Name&#39;)
graph_data &lt;- graph_data[ARG != &#39;spike&#39;]
graph_data &lt;- graph_data[!(ARG %in% c(&#39;16S&#39;))]
graph_data &lt;- merge(metadata[,c(1,2,4)], graph_data, by.x = &#39;ID&#39;, by.y = &#39;Sample_Name&#39;)
graph_data &lt;- graph_data[, sum(value), by = c(&#39;Sample&#39;, &#39;Matrix&#39;, &#39;ARG_Family&#39;, &#39;ARG&#39;, &#39;Primer&#39;)]

set(graph_data, j = classification, value = factor(graph_data[[classification]], levels = c(&#39;spike&#39;, rev(unique(graph_data[[classification]][!(graph_data[[classification]]==&#39;spike&#39;)])))))
setkey(graph_data, &quot;ARG&quot;, &quot;Sample&quot;)
set(graph_data, which(is.na(graph_data[[&#39;Day&#39;]])), j=&#39;Day&#39;, value=0)

graph_data &lt;- rbindlist(list(
  cbind(graph_data[,c(&#39;Sample&#39;, &#39;ARG&#39;, &#39;ARG_Family&#39;, &#39;Matrix&#39;, &#39;V1&#39;)], Tech = &#39;DARTE-QM&#39;), 
  cbind(SC_I[,c(&#39;Sample&#39;, &#39;ARG&#39;, &#39;ARG_Family&#39;, &#39;Matrix&#39;, &#39;V1&#39;)], Tech = &#39;Metagenome&#39;)), fill = TRUE)
graph_data[, relative_abundance := round(V1/sum(V1), 4), by = c(&#39;Sample&#39;)]
set(graph_data, j = &#39;Matrix&#39;, value = as.character(graph_data$Matrix))
set(graph_data, which(graph_data$Matrix == &quot;manure&quot;), &#39;Matrix&#39;, &quot;Swine Manure-A&quot;)
set(graph_data, which(graph_data$Matrix == &quot;effluent&quot;), &#39;Matrix&#39;, &quot;Effluent&quot;)
set(graph_data, which(graph_data$Matrix == &quot;Manure-Soil A&quot;), &#39;Matrix&#39;, &quot;Soil-A + Manure-A&quot;)

graph_data &lt;- graph_data[!(is.na(relative_abundance))]
graph_data &lt;- graph_data[graph_data$Matrix != &#39;Effluent&#39;,]
graph_data &lt;- graph_data[V1 != 0]
graph_data[, relative_abundance := round(V1/sum(V1), 4), by = c(&#39;Sample&#39;)]

classification_order &lt;- setorder(graph_data[, lapply(.SD, sum, na.rm=TRUE), by=ARG_Family, .SDcols=c(&quot;relative_abundance&quot;)], -relative_abundance)
classification_order &lt;- classification_order[round(classification_order[[2]]*100 / length(unique(graph_data[Tech == &quot;DARTE-QM&quot;]$Sample)),1) &gt;= 0.5,]

graph_data &lt;- graph_data[ARG_Family %in% classification_order$ARG_Family]
set(graph_data, j = &#39;ARG_Family&#39;, value = factor(graph_data$ARG_Family, levels = classification_order[[1]]))

missing &lt;- unique(sort(as.character(graph_data[Tech != &quot;DARTE-QM&quot;]$ARG_Family[!(graph_data[Tech != &quot;DARTE-QM&quot;]$ARG_Family %in% graph_data[Tech == &quot;DARTE-QM&quot;]$ARG_Family)])))
setorder(graph_data, &#39;ARG_Family&#39;)
genes &lt;- as.character(graph_data$ARG_Family)
for(i in seq_along(missing)){
  genes[genes %in% missing[i]] &lt;- c(&quot;ceo&quot;, &quot;mex&quot;, &quot;mux&quot;, &quot;ole&quot;, &quot;bla&quot;, &quot;rph&quot;, &quot;bla&quot;, &quot;tlr&quot;, &quot;Van&quot;)[i]
}
set(graph_data, j = &#39;ARG_Family&#39;, value = factor(genes, levels = unique(genes)))
setkey(graph_data, &#39;Sample&#39;)
set(graph_data, j = &#39;Matrix&#39;, value = factor(graph_data$Matrix, levels = unique(metadata$Matrix)))

graph_data &lt;- graph_data[, lapply(.SD, sum, na.rm=TRUE), by=c(&#39;Sample&#39;, &quot;ARG&quot;, &quot;ARG_Family&quot;, &#39;Matrix&#39;, &#39;Tech&#39;), 
                         .SDcols=c(&quot;V1&quot;, &quot;relative_abundance&quot;) ] 
graph_data[, relative_abundance := round(V1/sum(V1), 4), by = c(&#39;Sample&#39;)]
graph_colors &lt;- schuylR::create_palette(length(unique(graph_data[[&#39;ARG_Family&#39;]])))

gene_classes &lt;- sort(unique(graph_data[[&#39;ARG_Family&#39;]]))
compare_data &lt;- graph_data[c(grep(&#39;Swine Manure-A&#39;, graph_data$Sample), grep(&#39;Soil-A&#39;, graph_data$Sample), grep(&#39;_L0&#39;, graph_data$Sample)), ]
compare_data &lt;- compare_data[Matrix %in% c(&#39;Swine Manure-A&#39;, &#39;Soil-A + Manure-A&#39;)]

set(compare_data, j = &#39;Sample&#39;, value = factor(compare_data[[&#39;Sample&#39;]], levels = unique(graph_data[[&#39;Sample&#39;]])))
set(compare_data, j = &#39;Matrix&#39;, value = factor(compare_data[[&#39;Matrix&#39;]], levels = unique(graph_data[[&#39;Matrix&#39;]])))

profile_compare &lt;- ggplot(compare_data, 
       aes_string(x = &quot;Sample&quot;, 
                  y = &quot;relative_abundance&quot;,
                  fill = &quot;ARG_Family&quot;)) + 
  guides(fill = guide_legend(ncol = 2)) + 
  scale_fill_manual(
    values = graph_colors[gene_classes[gene_classes %in% unique(compare_data[[&#39;ARG_Family&#39;]])]],
    aesthetics = c(&quot;color&quot;, &quot;fill&quot;)
  ) + 
  facet_nested(.~ Tech + Matrix, scales = &quot;free&quot;, space = &quot;free&quot;) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;, 
           size = 0.001, width = 0.95, color = &#39;black&#39;) + 
  ylab(&quot;Relative Abundance&quot;) + 
  theme_bw() + 
  theme(axis.text.x = element_text(size= 12, angle = 330, hjust = -0.05), 
         axis.text.y = element_blank(), 
         axis.title.x = element_blank(), 
         axis.title.y = element_text(size = 12, face = &quot;bold&quot;), 
         axis.ticks.x = element_blank(), 
         legend.title = element_text(size = 12, face = &quot;bold&quot;), 
         legend.text = element_text(size = 12), 
         legend.spacing.x = unit(0.005, &quot;npc&quot;), 
         legend.key.size = unit(6, &quot;mm&quot;), 
         panel.background = element_rect(color = &quot;black&quot;, size = 1.5), 
         panel.spacing = unit(0.01, &quot;npc&quot;), 
         strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
         strip.background = element_rect(colour = &quot;black&quot;, size = 1.4, fill = &quot;white&quot;),
         panel.grid.major.x = element_blank()) + 
  scale_y_continuous(expand = expansion(mult = c(0.0037, 0.003), add = c(0, 0))) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.51))</code></pre>
<p><img src="metagenome_comparison_files/figure-html/shotgun_profile-1.png" width="1152" /> <br> <br></p>
</div>
<div id="diversity" class="section level3">
<h3>Diversity</h3>
<pre class="r"><code>index = &quot;shannon&quot;
colors = &quot;default&quot;
alpha &lt;- compare_data
alpha &lt;- dcast(alpha, ARG ~ Sample, value.var = &#39;relative_abundance&#39;, fun.aggregate = sum)
alpha &lt;- alpha[,-1]
replace_DT_NA(alpha)

alpha &lt;- alpha[,lapply(.SD,function(sample) sample/sum(sample))]
alpha &lt;- -alpha * log(alpha)
alpha &lt;- alpha[,lapply(.SD, sum, na.rm = TRUE)]

samples &lt;- dcast(compare_data, Sample + Tech + Matrix ~ ARG, value.var = &#39;relative_abundance&#39;)

alpha_data &lt;- data.table(Sample = samples$Sample, 
    Matrix = samples$Matrix,
    Tech = samples$Tech,
    Alpha = unlist(alpha))  
color_count &lt;- length(unique(paste(alpha_data$Tech, alpha_data$Matrix)))
alpha_graph_colors &lt;- schuylR::create_palette(color_count, colors)
  
div_graph &lt;- ggplot(alpha_data, aes(Matrix, Alpha)) + 
    geom_boxplot(show.legend = FALSE, fill = alpha_graph_colors) + 
    theme_bw() +   
  theme(
    axis.text.x = element_text(size = 12, face = &quot;bold&quot;), 
    axis.text.y = element_text(hjust = 0.95, size = 12), 
    axis.title.x = element_blank(), 
    axis.title.y = element_text(size = 12, face = &quot;bold&quot;), 
    axis.ticks.x = element_blank(), 
    strip.text.x = element_text(size = 12, face = &quot;bold&quot;), 
    strip.background = element_rect(colour = &quot;black&quot;, size = 1.4, fill = &quot;white&quot;),
    panel.grid.major.x = element_blank()) + 
    labs(y = paste(&quot;Alpha-Diverity (&quot;, stringr::str_to_title(index), 
      &quot; Index)&quot;, sep = &quot;&quot;)) + 
    facet_nested(.~ Tech, scales = &quot;free&quot;, space = &quot;free&quot;)  +
  scale_y_continuous(breaks = seq(2, 4, 0.2), limits = c(2,4))</code></pre>
<p><img src="metagenome_comparison_files/figure-html/div_graph-1.png" width="480" /> <br> <br></p>
</div>
<div id="resitome-differences" class="section level3">
<h3>Resitome Differences</h3>
<div id="swine-manure-a" class="section level4">
<h4>Swine Manure-A</h4>
<p>DARTE-QM had 128 ARGs in Swine Manure-A samples, with an alpha-diversity measured at 3.37. <br></p>
<p>compared to metagenomes which had 84 ARGs in Swine Manure-A samples, with an alpha-diversity measured at 3.18. <br></p>
<p>The metagenome found 47 ARGs that were not detected with DARTE-QM.</p>
<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName">
Show Table
</button>
<div id="BlockName" class="collapse">
<div id="htmlwidget-dcafb6c34209ecab17ec" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dcafb6c34209ecab17ec">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47"],["tetT","tetA(P)","tet(36)","tetM","tet(32)","tetW","tetY","tetZ","tet(40)","tet(33)","tetB(P)","tet(31)","tetG","tetS","tetQ","tetX","APH(3')-Ia","APH(2'')-Ib","ANT(6)-Ib","APH(3'')-Ib","APH(3')-VI","sul1","sul2","lnuA","lnuF","lnuC","ErmG","ErmB","ErmT","ErmC","ErmF","ErmQ","ErmA","ErmX","Erm(47)","Erm(35)","ErmY","Erm(42)","Erm(36)","ANT(6)-Ia","aadA7","aadA5","mdtO","str","APH(4)-Ia","aadA24","mefA"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ARG<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br></p>
</div>
<div id="manure-soil" class="section level4">
<h4>Manure-Soil</h4>
<p>DARTE-QM had 128 ARGs in Manure-Soil samples, with an alpha-diversity measured at 3.38. <br></p>
<p>compared to metagenomes which had 139 ARGs in Swine Manure-A samples, with an alpha-diversity measured at 3.4. <br></p>
<p>The metagenome found 62 ARGs that were not detected with DARTE-QM.</p>
<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName2">
Show Table
</button>
<div id="BlockName2" class="collapse">
<div id="htmlwidget-36cd0a3bd2dc61afe388" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-36cd0a3bd2dc61afe388">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62"],["AAC(3)-IVa","AAC(6')-Ib11","aadA15","aadA24","aadA5","aadA7","ACI-1","acrB","ANT(6)-Ia","ANT(6)-Ib","APH(2'')-Ib","APH(3'')-Ib","APH(3')-Ia","APH(3')-IIa","APH(3')-IIIa","APH(4)-Ia","cfr(C)","Erm(35)","Erm(36)","Erm(42)","Erm(47)","ErmA","ErmB","ErmC","ErmF","ErmG","ErmQ","ErmT","ErmX","lnuA","lnuC","lnuD","lnuF","mdsB","mdtA","mdtB","mdtC","mdtO","mefA","mefB","OXA-347","str","sul1","sul2","tet(31)","tet(33)","tet(36)","tet(40)","tetA","tetA(P)","tetB(P)","tetC","tetG","tetL","tetM","tetQ","tetS","tetT","tetW","tetX","tetY","tetZ"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ARG<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br> <br></p>
</div>
</div>
</div>
<div id="darte-qm-target-args" class="section level2 tabset tabset-pills">
<h2>DARTE-QM Target ARGs</h2>
<div id="profiles-1" class="section level3">
<h3>Profiles</h3>
<pre class="r"><code>compare_data &lt;- compare_data[ARG %in% classifications$ARG]
compare_data[, relative_abundance := round(V1/sum(V1), 4), by = c(&#39;Sample&#39;)]

profile_compare &lt;- ggplot(compare_data, 
       aes_string(x = &quot;Sample&quot;, 
                  y = &quot;relative_abundance&quot;,
                  fill = &quot;ARG_Family&quot;)) + 
  guides(fill = guide_legend(ncol = 2)) + 
  scale_fill_manual(
    values = graph_colors[gene_classes[gene_classes %in% unique(compare_data[[&#39;ARG_Family&#39;]])]],
    aesthetics = c(&quot;color&quot;, &quot;fill&quot;)
  ) + 
  facet_nested(.~ Tech + Matrix, scales = &quot;free&quot;, space = &quot;free&quot;) +
  geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;, 
           size = 0.005, width = 0.95, color = &#39;black&#39;) + 
  ylab(&quot;Relative Abundance&quot;) + 
  theme_bw() + 
  theme(axis.text.x = element_text(size= 12, angle = 330, hjust = -0.05), 
         axis.text.y = element_blank(), 
         axis.title.x = element_blank(), 
         axis.title.y = element_text(size = 12, face = &quot;bold&quot;), 
         axis.ticks.x = element_blank(), 
         legend.title = element_text(size = 12, face = &quot;bold&quot;), 
         legend.text = element_text(size = 12), 
         legend.spacing.x = unit(0.005, &quot;npc&quot;), 
         legend.key.size = unit(6, &quot;mm&quot;), 
         panel.background = element_rect(color = &quot;black&quot;, size = 1.5), 
         panel.spacing = unit(0.01, &quot;npc&quot;), 
         strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
         strip.background = element_rect(colour = &quot;black&quot;, size = 1.4, fill = &quot;white&quot;),
         panel.grid.major.x = element_blank()) + 
  scale_y_continuous(expand = expansion(mult = c(0.0037, 0.003), add = c(0, 0))) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.51))</code></pre>
<p><img src="metagenome_comparison_files/figure-html/target_profile-1.png" width="1152" /> <br> <br></p>
</div>
<div id="diversity-1" class="section level3">
<h3>Diversity</h3>
<pre class="r"><code>index = &quot;shannon&quot;
colors = &quot;default&quot;
alpha &lt;- compare_data
alpha &lt;- dcast(alpha, ARG ~ Sample, value.var = &#39;relative_abundance&#39;, fun.aggregate = sum)
alpha &lt;- alpha[,-1]
replace_DT_NA(alpha)

alpha &lt;- alpha[,lapply(.SD,function(sample) sample/sum(sample))]
alpha &lt;- -alpha * log(alpha)
alpha &lt;- alpha[,lapply(.SD, sum, na.rm = TRUE)]

samples &lt;- dcast(compare_data, Sample + Tech + Matrix ~ ARG, value.var = &#39;relative_abundance&#39;)

alpha_data &lt;- data.table(Sample = samples$Sample, 
    Matrix = samples$Matrix,
    Tech = samples$Tech,
    Alpha = unlist(alpha))
  color_count &lt;- length(unique(paste(alpha_data$Tech, alpha_data$Matrix)))
  graph_colors &lt;- create_palette(color_count, colors)
  
div_graph &lt;- ggplot(alpha_data, aes(Matrix, Alpha)) + 
    geom_boxplot(show.legend = FALSE, fill = graph_colors) + 
    theme_bw() + 
  theme(
    axis.text.x = element_text(size = 12, face = &quot;bold&quot;), 
    axis.text.y = element_text(hjust = 0.95, size = 12), 
    axis.title.x = element_blank(), 
    axis.title.y = element_text(size = 12, face = &quot;bold&quot;), 
    axis.ticks.x = element_blank(), 
    strip.text.x = element_text(size = 12, face = &quot;bold&quot;), 
    strip.background = element_rect(colour = &quot;black&quot;, size = 1.4, fill = &quot;white&quot;),
    panel.grid.major.x = element_blank()) + 
    labs(y = paste(&quot;Alpha-Diverity (&quot;, stringr::str_to_title(index), 
      &quot; Index)&quot;, sep = &quot;&quot;)) + 
    facet_nested(.~ Tech, scales = &quot;free&quot;, space = &quot;free&quot;)  +
  scale_y_continuous(breaks = seq(2, 4, 0.2), limits = c(2,4))</code></pre>
<p><img src="metagenome_comparison_files/figure-html/div_graph2-1.png" width="480" /> <br> <br></p>
</div>
<div id="resitome-differences-1" class="section level3">
<h3>Resitome Differences</h3>
<div id="swine-manure-a-1" class="section level4">
<h4>Swine Manure-A</h4>
<p>With the ARGs filtered to only include those targeted by DARTE-QM, metagenomes found 73 of the 128 that DARTE-QM found, with an alpha diversity of 3.06. The metagenome technology was able to find 26 ARGs targeted by DARTE-QM that the DARTE-QM primers were unable to successfully capture.</p>
<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName3">
Show Table
</button>
<div id="BlockName3" class="collapse">
<div id="htmlwidget-50b766226a687439895d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-50b766226a687439895d">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["tetO","tetV","tetJ","tetD","tet(45)","APH(2'')-If","APH(6)-Ia","lnuB","Erm(33)","ANT(4')-Ib","AAC(6')-Ip","aadA8","aadA22","aadA1","aadA16","lsaC","mphG","AAC(3)-Ib/AAC(6')-Ib''","AAC(6')-Ib11","AAC(6')-Iid","AAC(6')-Iak","AAC(3)-Ib","AAC(3)-IIc","OXA-258","aadA11","mphB"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ARG<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br></p>
</div>
<div id="manunre-soil" class="section level4">
<h4>Manunre-Soil</h4>
<p>With the ARGs filtered to only include those targeted by DARTE-QM, metagenomes found 106 of the 128, with an alpha diversity of 3.12. The metagenome technology was able to find 44 ARGs targeted by DARTE-QM that the DARTE-QM primers were unable to successfully capture.</p>
<button class="btn btn-primary" data-toggle="collapse" data-target="#BlockName4">
Show Table
</button>
<div id="BlockName4" class="collapse">
<div id="htmlwidget-14a0a82a9e87ea578466" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-14a0a82a9e87ea578466">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44"],["tet(32)","tetD","tetO","tetV","tet(45)","tetJ","tet(39)","APH(6)-Ia","APH(2'')-If","APH(3')-VI","lnuB","Erm(33)","ErmY","ANT(4')-Ib","aadA22","aadA1","AAC(6')-Ip","aadA8","aadA6","aadA14","aadA16","aadA11","aadA13","acrA","lsaC","mphB","mphD","AAC(3)-Ib/AAC(6')-Ib''","AAC(6')-Iak","AAC(3)-IIc","AAC(6')-Iid","OXA-258","otrC","lsaB","OXA-29","AAC(3)-Ib","tet(43)","APH(3')-XV","tet(42)","APH(3')-VIa","otrB","AAC(2')-Ib","tetA(46)","OXA-209"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ARG<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<p><br></p>
</div>
</div>
</div>

<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
