<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Schuyler D. Smith" />

<meta name="date" content="2021-04-15" />

<title>Mock-Community Analysis</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.17/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DARTE-QM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li>
  <a href="sequencing.html">
    <span class="fas fa-dna"></span>
     
    Sequencing
  </a>
</li>
<li>
  <a href="primer_analysis.html">
    <span class="fas fa-barcode"></span>
     
    Primers
  </a>
</li>
<li>
  <a href="mock_analysis.html">
    <span class="fas fa-vial"></span>
     
    Mock
  </a>
</li>
<li>
  <a href="sample_analysis.html">
    <span class="fas fa-chart-pie"></span>
     
    Sample
  </a>
</li>
<li>
  <a href="metagenome_comparison.html">
    <span class="fas fa-greater-than-equal"></span>
     
    Metagenome
  </a>
</li>
<li>
  <a href="data.html">
    <span class="fas fa-database"></span>
     
    Data
  </a>
</li>
<li>
  <a href="packages_and_functions.html">
    <span class="fas fa-box-open"></span>
     
    Packages
  </a>
</li>
<li>
  <a href="https://github.com/schuyler-smith/DARTE-QM/">
    <span class="fas fa-code"></span>
     
    Source Code
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/schuyler-smith">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://germslab.org/">
    <span class="fas fa-flask"></span>
     
  </a>
</li>
<li>
  <a href="https://www.bcb.iastate.edu/people/schuyler-smith">
    <span class="fas fa-university"></span>
     
  </a>
</li>
<li>
  <a href="mailto:sdsmith@iastate.edu">
    <span class="fas fa-envelope fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Mock-Community Analysis</h1>
<h4 class="author"><a href="https://schuyler-smith.github.io/" target="_blank" >Schuyler D. Smith</a></h4>
<h4 class="date">2021-04-15</h4>

</div>


<p><br> <br></p>
<div id="theoretical-arg-counts" class="section level2 tabset tabset-pills">
<h2>Theoretical ARG Counts</h2>
<div id="arg-abundances" class="section level3">
<h3>ARG Abundances</h3>
<pre class="r"><code>mock_samples &lt;- as.character(metadata[grep(&#39;Mock&#39;, metadata$`ID`),][`Passed QC` == TRUE][[&quot;Sample&quot;]])
for_blast &lt;- data.table::fread(&#39;../data/mock/forward_primers.blast&#39;)
  for_blast &lt;- merge(for_blast, classifications, by.x = &quot;V1&quot;, by.y = &quot;Primer&quot;)
  for_blast &lt;- for_blast[ARG != &quot;16S&quot;]
  for_blast &lt;- for_blast[V3 == 100]
  for_blast &lt;- for_blast[V4 &gt;= 18]
rev_blast &lt;- data.table::fread(&#39;../data/mock/reverse_primers.blast&#39;)
  rev_blast &lt;- merge(rev_blast, classifications, by.x = &quot;V1&quot;, by.y = &quot;Primer&quot;)
  rev_blast &lt;- rev_blast[ARG != &quot;16S&quot;]
  rev_blast &lt;- rev_blast[V3 == 100]
  rev_blast &lt;- rev_blast[V4 &gt;= 18]
mock_blast &lt;- for_blast[paste0( for_blast[[1]], for_blast[[2]]) %in% paste0(rev_blast[[1]], rev_blast[[2]]),]

strains[, Abundance := Genome_copies_per_microgram * micrograms]
mock_genome &lt;- vector()
i=1
for(genome in mock_blast$V2){
    genome &lt;- mock_genome_names[sapply(strsplit(mock_genome_names, &#39; &#39;), FUN=function(x){any(x %in% genome)})]
    genome &lt;- strains[apply(strains,1,FUN=function(x){any(x %in% unlist(strsplit(genome, &quot; &quot;)))})]$Strain
    mock_genome &lt;- c(mock_genome, genome)
} 
set(mock_blast, j = &quot;V2&quot;, value = mock_genome)
setorder(mock_blast, V2, V9)

mock_expected_counts &lt;- mock_blast[,c(&#39;V2&#39;, &#39;V1&#39;, &#39;V9&#39;, &#39;ARG&#39;, &#39;ARG_Family&#39;)]
setnames(mock_expected_counts, c(&#39;V2&#39;,&#39;V1&#39;), c(&#39;Microbe&#39;, &#39;Primer&#39;))
mock_expected_counts &lt;- merge(mock_expected_counts, strains, by.x = &quot;Microbe&quot;, by.y = &quot;Strain&quot;)
mock_expected_counts &lt;- mock_expected_counts[, .(count = sum(Abundance)), by = c( &#39;ARG_Family&#39;, &quot;ARG&quot;, &#39;Primer&#39;, &quot;Microbe&quot;)]
mock_expected_counts[, count := round(count/sum(count), 4)]
mock_expected_counts[,Sample := &#39;Theoretical&#39;]
setcolorder(mock_expected_counts, c(&quot;Sample&quot;, &quot;Microbe&quot;, &quot;ARG&quot;, &quot;ARG_Family&quot;, &#39;Primer&#39;, &quot;count&quot;))</code></pre>
<div id="htmlwidget-469cc475f62b40f912e0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-469cc475f62b40f912e0">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85"],["Cloacibacillus porcorum CL-84","Cloacibacillus porcorum CL-84","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Escherichia coli mg1655","Parabacteroides merdae","Parabacteroides merdae","Parabacteroides merdae","Roseburia hominis A2-183","Roseburia hominis A2-183","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Salmonella enterica serovar Typhimurium SL1344","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300","Staphylococcus aureus USA300"],["tetO","lnuC","APH(3'')","APH(3'')","acrB","acrA","emrE","mdfA","mdtG","mdtH","MdtK","pmrE","mdtA","mdtB","mdtC","MdtD","PmrF","arnA","emrY","emrK","acrD","emrR","emrA","emrB","TolC","AcrS","AcrE","AcrF","mdtE","mdtF","emrD","mdtL","mdtP","mdtO","mdtN","pmrB","pmrA","pmrC","mdtM","catA","tetQ","tetQ","APH(3'')","tetW","sul2","sul2","sul2","APH(6)","APH(3'')","mdsC","mdsB","mdsA","acrB","acrA","mdfA","mdtG","mdtH","MdtK","AAC(6')","pmrE","mdtB","mdtC","mdtD","PmrF","acrD","emrR","emrA","emrB","TolC","AcrE","AcrF","emrD","pmrB","pmrA","mdtM","APH(3'')","blaZ","mphC","msrA","mecA","mecA","tet","mepA","mprF","sav1866"],["tet","lnu","aph","aph","acr","acr","emr","mdf","mdt","mdt","mdt","pmr","mdt","mdt","mdt","mdt","pmr","arn","emr","emr","acr","emr","emr","emr","tol","acr","acr","acr","mdt","mdt","emr","mdt","mdt","mdt","mdt","pmr","pmr","pmr","mdt","cat","tet","tet","aph","tet","sul","sul","sul","aph","aph","mds","mds","mds","acr","acr","mdf","mdt","mdt","mdt","aac","pmr","mdt","mdt","mdt","pmr","acr","emr","emr","emr","tol","acr","acr","emr","pmr","pmr","mdt","aph","bla","mph","msr","mec","mec","tet","mep","mpr","amr"],["tet_O_W_O_-1_1_AY196921","lnu_C__1_AY928180","aph_3__-IIa_1_X57709","aph_3__-Ia_1_V00359","acrB_F_DARTE-QM-53_R_DARTE-QM-53","acrA_F_DARTE-QM-55_R_DARTE-QM-55","emrE_F_DARTE-QM-71_R_DARTE-QM-71","mdfA_F_DARTE-QM-72_R_DARTE-QM-72","mdtG_F_DARTE-QM-92_R_DARTE-QM-92","mdtH_F_DARTE-QM-100_R_DARTE-QM-100","mdtK_F_DARTE-QM-98_R_DARTE-QM-98","PmrE_F_DARTE-QM-80_R_DARTE-QM-80","mdtA_F_DARTE-QM-91_R_DARTE-QM-91","mdtB_F_DARTE-QM-89_R_DARTE-QM-89","mdtC_F_DARTE-QM-87_R_DARTE-QM-87","mdtD_F_DARTE-QM-96_R_DARTE-QM-96","PmrF_F_DARTE-QM-78_R_DARTE-QM-78","arnA_F_DARTE-QM-107_R_DARTE-QM-107","emrY_F_DARTE-QM-46_R_DARTE-QM-46","emrK_F_DARTE-QM-109_R_DARTE-QM-109","acrD_F_DARTE-QM-59_R_DARTE-QM-59","emrR_F_DARTE-QM-49_R_DARTE-QM-49","emrA_F_DARTE-QM-65_R_DARTE-QM-65","emrB_F_DARTE-QM-67_R_DARTE-QM-68","tolC_F_DARTE-QM-63_R_DARTE-QM-63","acrS_F_DARTE-QM-52_R_DARTE-QM-52","acrE_F_DARTE-QM-61_R_DARTE-QM-61","acrF_F_DARTE-QM-57_R_DARTE-QM-57","mdtE_F_DARTE-QM-95_R_DARTE-QM-95","mdtF_F_DARTE-QM-94_R_DARTE-QM-94","emrD_F_DARTE-QM-69_R_DARTE-QM-69","mdtL_F_DARTE-QM-105_R_DARTE-QM-105","mdtP_F_DARTE-QM-84_R_DARTE-QM-84","mdtO_F_DARTE-QM-74_R_DARTE-QM-74","mdtN_F_DARTE-QM-102_R_DARTE-QM-102","PmrB_F_DARTE-QM-110_R_DARTE-QM-110","PmrA_F_DARTE-QM-75_R_DARTE-QM-75","PmrC_F_DARTE-QM-108_R_DARTE-QM-108","mdtM_F_DARTE-QM-103_R_DARTE-QM-103","catA1_1_V00622","tet_Q__1_L33696","tet_Q__2_X58717","aph_3__-IIIa_1_AF330699","tet_W__1_DQ060146","sul2_17_U57647","sul2_F_DARTE-QM-19_R_DARTE-QM-18","sul2_F_DARTE-QM-19_R_DARTE-QM-19","aph_6_-Id_1_M28829","aph_3___-Ib_3_AF321550","mdsC_F_DARTE-QM-48_R_DARTE-QM-48","mdsB_F_DARTE-QM-47_R_DARTE-QM-47","mdsA_F_DARTE-QM-51_R_DARTE-QM-51","acrB_F_DARTE-QM-54_R_DARTE-QM-54","acrA_F_DARTE-QM-56_R_DARTE-QM-56","mdfA_F_DARTE-QM-73_R_DARTE-QM-73","mdtG_F_DARTE-QM-93_R_DARTE-QM-93","mdtH_F_DARTE-QM-101_R_DARTE-QM-101","mdtK_F_DARTE-QM-99_R_DARTE-QM-99","aac_6__-Iaa_1_NC_003197","PmrE_F_DARTE-QM-81_R_DARTE-QM-81","mdtB_F_DARTE-QM-90_R_DARTE-QM-90","mdtC_F_DARTE-QM-88_R_DARTE-QM-88","mdtD_F_DARTE-QM-97_R_DARTE-QM-97","PmrF_F_DARTE-QM-79_R_DARTE-QM-79","acrD_F_DARTE-QM-60_R_DARTE-QM-60","emrR_F_DARTE-QM-50_R_DARTE-QM-50","emrA_F_DARTE-QM-66_R_DARTE-QM-66","emrB_F_DARTE-QM-68_R_DARTE-QM-68","tolC_F_DARTE-QM-64_R_DARTE-QM-64","acrE_F_DARTE-QM-62_R_DARTE-QM-62","acrF_F_DARTE-QM-58_R_DARTE-QM-58","emrD_F_DARTE-QM-70_R_DARTE-QM-70","PmrB_F_DARTE-QM-111_R_DARTE-QM-111","PmrA_F_DARTE-QM-76_R_DARTE-QM-76","mdtM_F_DARTE-QM-104_R_DARTE-QM-104","aph_3__-IIIa_1_AF330699","blaZ_36_AJ400722","mph_C__2_AF167161","msr_A__2_AB013298","mecA_10_AB512767","mecA_F_DARTE-QM-86_R_DARTE-QM-86","tet38_F_DARTE-QM-82_R_DARTE-QM-82","mepA_F_DARTE-QM-85_R_DARTE-QM-85","mprF_F_DARTE-QM-77_R_DARTE-QM-77","sav1866_F_DARTE-QM-106_R_DARTE-QM-106"],[0.024,0.024,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.024,0.024,0.024,0.018,0.018,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0103,0.0144,0.0144,0.0144,0.0144,0.0144,0.0144,0.0144,0.0144,0.0144,0.0144]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Microbe<\/th>\n      <th>ARG<\/th>\n      <th>ARG_Family<\/th>\n      <th>Primer<\/th>\n      <th>count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br> <br></p>
</div>
<div id="profile" class="section level3">
<h3>Profile</h3>
<pre class="r"><code>graph_data &lt;- mock_expected_counts[, c(&quot;Sample&quot;, &quot;ARG_Family&quot;, &quot;count&quot;)]
graph_data &lt;- graph_data[, lapply(.SD, sum, na.rm=TRUE), by=ARG_Family, .SDcols=c(&quot;count&quot;)]
set(graph_data, j = &quot;ARG_Family&quot;, value = factor(graph_data$ARG_Family, levels = sort(graph_data$ARG_Family)))
setorder(graph_data, -ARG_Family)
graph_data[, ypos := cumsum(count)- 0.5*count]
mock_profile &lt;- ggplot(graph_data, aes(x=&quot;&quot;, y=count, fill=ARG_Family)) +
  geom_bar(stat=&quot;identity&quot;, width=1) +
  coord_polar(&quot;y&quot;, start=0) +
  theme_void() + 
  scale_fill_manual(values = schuylR::create_palette(length(unique(graph_data[[&#39;ARG_Family&#39;]])))) +
  geom_text(aes(y = ypos, label = ARG_Family), color = &quot;black&quot;, size=4, nudge_x = 0.3)</code></pre>
<p><img src="mock_analysis_files/figure-html/theoretical_pie_out-1.png" width="672" /> <br> <br></p>
</div>
</div>
<div id="primer-success" class="section level2 tabset tabset-pills">
<h2>Primer Success</h2>
<pre class="r"><code>mock_expected_ARGs &lt;- c(unique(sort(mock_expected_counts$Primer)), &#39;spike&#39;)
mock_observed_counts &lt;- read_counts[, c(1:3, grep(&#39;Mock&#39;, colnames(read_counts))), with = FALSE]
mock_observed_counts &lt;- mock_observed_counts[rowSums(mock_observed_counts[,-c(1:3)]) &gt; 1,]
mock_successful_primers &lt;- mock_expected_ARGs[mock_expected_ARGs %in% mock_observed_counts$Primer]
unexpected_primers &lt;- mock_observed_counts$Primer[!(mock_observed_counts$Primer %in% mock_expected_ARGs)]
setnames(mock_observed_counts, colnames(mock_observed_counts)[-c(1:3)], 
         as.character(metadata$Sample[match(colnames(mock_observed_counts)[-c(1:3)], metadata$ID)]))</code></pre>
<p><br></p>
<div id="arg-heatmap" class="section level3">
<h3>ARG-Heatmap</h3>
<pre class="r"><code>classification &lt;- &#39;ARG&#39;
expected_genes &lt;- mock_expected_counts[,c(&#39;Sample&#39;, classification, &#39;count&#39;), with=FALSE]
expected_genes[, Abundance := count * 20000]
expected_genes &lt;- dcast(expected_genes, ARG ~ Sample, value.var = &#39;Abundance&#39;, fun.aggregate = sum)

graph_data &lt;- mock_observed_counts[get(classification) %in% c(mock_expected_counts[[classification]], &quot;spike&quot;)]
graph_data &lt;- merge(expected_genes, graph_data[,-c(1:3)[!(colnames(graph_data)[1:3] %in% classification)], 
                                               with=FALSE], by = classification, all = TRUE)
graph_data &lt;- schuylR::replace_NA(graph_data)
graph_data &lt;- rarefy(data=graph_data, min(colSums(graph_data[,-1])))
graph_data &lt;- melt(graph_data, id.vars = classification, variable.name = &#39;Sample&#39;, value.name = &#39;Abundance&#39;)
set(graph_data, j = &#39;Sample&#39;, value = factor(graph_data[[&#39;Sample&#39;]], 
                                               levels = c(&#39;Theoretical&#39;, mock_samples)))
graph_data &lt;- graph_data[, lapply(.SD, sum, na.rm=TRUE), by = c(&#39;Sample&#39;, classification), .SDcols=c(&quot;Abundance&quot;)]

set(graph_data, j = classification, value = factor(as.character(graph_data[[classification]]), 
                                                   levels = rev(c(sort(unique(graph_data[get(classification) != &#39;spike&#39;][[classification]])), &#39;spike&#39;))))
set(graph_data, which(graph_data[[&#39;Abundance&#39;]] &lt; 1), j=&#39;Abundance&#39;, value=NA)
set(graph_data, j = &quot;Abundance&quot;, value = log(graph_data$Abundance, base = 2))
ARG_counts &lt;- ggplot(graph_data, aes_string(&quot;Sample&quot;, classification, fill = &quot;Abundance&quot;)) + 
  geom_tile(color = &quot;white&quot;, size = 0.25) +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 20, hjust = 1, size = 10), 
        axis.text.y = element_text(hjust = 0.95, size = 10), 
        axis.title.x = element_text(size = 10, face = &quot;bold&quot;), 
        axis.title.y = element_text(size = 10, face = &quot;bold&quot;),
        axis.ticks.x = element_blank(), 
        legend.title = element_text(size = 8, face = &quot;bold&quot;), 
        legend.text = element_text(size = 8),
        legend.spacing.x = unit(0.005, &quot;npc&quot;), 
        legend.key.size = unit(6, &quot;mm&quot;), 
        legend.background = element_rect(fill = (alpha = 0), color = &quot;black&quot;, size = 0.25), 
        panel.background = element_rect(color = &quot;black&quot;,  size = 1.4), 
        strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
        strip.background = element_rect(colour = &quot;black&quot;, size = 1.4)) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.53)) +  
  ggraph::scale_fill_viridis(direction = -1, limits = c(min(graph_data$Abundance, na.rm = T), max(graph_data$Abundance, na.rm = T))) +
  labs(x = &quot;Sample&quot;)</code></pre>
<p>DARTE-QM found <strong>55</strong> (98.2%) of the <strong>56</strong> primers in the Mock-Community pangenomes that were targeted.</p>
<p><img src="mock_analysis_files/figure-html/ARG_counts_out-1.png" width="672" /> <br> <br></p>
</div>
<div id="primer-heatmap" class="section level3">
<h3>Primer-Heatmap</h3>
<p>DARTE-QM found <strong>80</strong> (94.1%) of the <strong>85</strong> primers in the Mock-Community pangenomes that were targeted.</p>
<pre class="r"><code>classification &lt;- &#39;Primer&#39;
expected_genes &lt;- mock_expected_counts[,c(&#39;Sample&#39;, classification, &#39;count&#39;), with=FALSE]
expected_genes[, Abundance := count * 20000]
expected_genes &lt;- dcast(expected_genes, Primer ~ Sample, value.var = &#39;Abundance&#39;, fun.aggregate = sum)

graph_data &lt;- mock_observed_counts[get(classification) %in% c(mock_expected_counts[[classification]], &quot;spike&quot;)]
graph_data &lt;- merge(expected_genes, graph_data[,-c(1:3)[!(colnames(graph_data)[1:3] %in% classification)], 
                                               with=FALSE], by = classification, all = TRUE)
graph_data &lt;- schuylR::replace_NA(graph_data)
graph_data &lt;- rarefy(data=graph_data, min(colSums(graph_data[,-1])))
graph_data &lt;- melt(graph_data, id.vars = classification, variable.name = &#39;Sample&#39;, value.name = &#39;Abundance&#39;)
set(graph_data, j = &#39;Sample&#39;, value = factor(graph_data[[&#39;Sample&#39;]], 
                                               levels = c(&#39;Theoretical&#39;, mock_samples)))
primer_graph_data &lt;- graph_data[, lapply(.SD, sum, na.rm=TRUE), by = c(&#39;Sample&#39;, classification), .SDcols=c(&quot;Abundance&quot;)]
setorder(primer_graph_data, Abundance)
set(primer_graph_data, j = classification, value = factor(as.character(primer_graph_data[[classification]]), 
                                                   levels = unique(as.character(primer_graph_data[[classification]]))))
set(primer_graph_data, which(primer_graph_data[[&#39;Abundance&#39;]] &lt; 1), j=&#39;Abundance&#39;, value=NA)

primer_graph_data$Primer &lt;- gsub(&#39;_.?_DARTE-QM&#39;, &#39;&#39;, primer_graph_data$Primer)
set(primer_graph_data, j = &#39;Primer&#39;, value = factor(primer_graph_data$Primer, levels = rev(gsub(&#39;_.?_DARTE-QM&#39;, &#39;&#39;, mock_expected_ARGs))))

Primer_counts &lt;- ggplot(primer_graph_data, aes_string(&quot;Sample&quot;, classification, fill = &quot;Abundance&quot;)) + 
  geom_tile(color = &quot;white&quot;, size = 0.25) +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 20, hjust = 1, size = 10), 
        axis.text.y = element_text(hjust = 0.95, size = 10), 
        axis.title.x = element_text(size = 10, face = &quot;bold&quot;), 
        axis.title.y = element_text(size = 10, face = &quot;bold&quot;),
        axis.ticks.x = element_blank(), 
        legend.title = element_text(size = 8, face = &quot;bold&quot;), 
        legend.text = element_text(size = 8),
        legend.spacing.x = unit(0.005, &quot;npc&quot;), 
        legend.key.size = unit(6, &quot;mm&quot;), 
        legend.background = element_rect(fill = (alpha = 0), color = &quot;black&quot;, size = 0.25), 
        panel.background = element_rect(color = &quot;black&quot;,  size = 1.4), 
        strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
        strip.background = element_rect(colour = &quot;black&quot;, size = 1.4)) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.53)) +  
  ggraph::scale_fill_viridis(limits = c(min(graph_data$Abundance, na.rm = T), max(graph_data$Abundance, na.rm = T))) +
  labs(x = &quot;Sample&quot;)</code></pre>
<p><img src="mock_analysis_files/figure-html/primer_counts_out-1.png" width="1152" /> <br> <br></p>
</div>
<div id="arg-family-heatmap" class="section level3">
<h3>ARG-Family-Heatmap</h3>
<pre class="r"><code>classification &lt;- &#39;ARG_Family&#39;
expected_genes &lt;- mock_expected_counts[,c(&#39;Sample&#39;, classification, &#39;count&#39;), with=FALSE]
expected_genes[, Abundance := count * 20000]
expected_genes &lt;- dcast(expected_genes, ARG_Family ~ Sample, value.var = &#39;Abundance&#39;, fun.aggregate = sum)

graph_data &lt;- mock_observed_counts[get(classification) %in% c(mock_expected_counts[[classification]], &quot;spike&quot;)]
graph_data &lt;- merge(expected_genes, graph_data[,-c(1:3)[!(colnames(graph_data)[1:3] %in% classification)], 
                                               with=FALSE], by = classification, all = TRUE)
graph_data &lt;- schuylR::replace_NA(graph_data)
graph_data &lt;- rarefy(data=graph_data, min(colSums(graph_data[,-1])))
graph_data &lt;- melt(graph_data, id.vars = classification, variable.name = &#39;Sample&#39;, value.name = &#39;Abundance&#39;)
set(graph_data, j = &#39;Sample&#39;, value = factor(graph_data[[&#39;Sample&#39;]], 
                                               levels = c(&#39;Theoretical&#39;, mock_samples)))
graph_data &lt;- graph_data[, lapply(.SD, sum, na.rm=TRUE), by = c(&#39;Sample&#39;, classification), .SDcols=c(&quot;Abundance&quot;)]

setorder(graph_data, Abundance)
set(graph_data, j = classification, value = factor(as.character(graph_data[[classification]]), levels = unique(as.character(graph_data[[classification]]))))
set(graph_data, which(graph_data[[&#39;Abundance&#39;]] &lt; 1), j=&#39;Abundance&#39;, value=NA)

setorder(graph_data, Sample)
ARG_Family_counts &lt;- ggplot(graph_data, aes_string(&quot;Sample&quot;, classification, fill = &quot;Abundance&quot;)) + 
  geom_tile(color = &quot;white&quot;, size = 0.25) +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 20, hjust = 1, size = 10), 
        axis.text.y = element_text(hjust = 0.95, size = 10), 
        axis.title.x = element_text(size = 10, face = &quot;bold&quot;), 
        axis.title.y = element_text(size = 10, face = &quot;bold&quot;),
        axis.ticks.x = element_blank(), 
        legend.title = element_text(size = 8, face = &quot;bold&quot;), 
        legend.text = element_text(size = 8),
        legend.spacing.x = unit(0.005, &quot;npc&quot;), 
        legend.key.size = unit(6, &quot;mm&quot;), 
        legend.background = element_rect(fill = (alpha = 0), color = &quot;black&quot;, size = 0.25), 
        panel.background = element_rect(color = &quot;black&quot;,  size = 1.4), 
        strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
        strip.background = element_rect(colour = &quot;black&quot;, size = 1.4)) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.53)) +  
  ggraph::scale_fill_viridis(limits = c(min(graph_data$Abundance, na.rm = T), max(graph_data$Abundance, na.rm = T))) +
  labs(x = &quot;Sample&quot;)</code></pre>
<p><img src="mock_analysis_files/figure-html/ARG_Family_counts_out-1.png" width="672" /> <br> <br></p>
</div>
</div>
<div id="distance-matrix" class="section level2">
<h2>Distance Matrix</h2>
<pre class="r"><code>dist_matrix &lt;- primer_graph_data
dist_matrix &lt;- dcast(dist_matrix[Primer %in% dist_matrix[, sum(Abundance), by = &#39;Primer&#39;][V1 &gt; 0]$Primer],
           Primer ~ Sample, value.var = &#39;Abundance&#39;)
replace_DT_NA(dist_matrix)
dist_matrix &lt;- as.matrix(round(vegdist(t(dist_matrix[,-c(1)])), 3))
dist_matrix_plot &lt;- dist_matrix
dist_matrix_plot[upper.tri(dist_matrix)] &lt;- NA
dist_matrix_plot &lt;- data.table(melt(dist_matrix_plot))
set(dist_matrix_plot, j = &quot;Var1&quot;, value = factor(dist_matrix_plot$Var1, levels = rev(unique(dist_matrix_plot$Var1))))
dist_matrix_plot &lt;- ggplot(data = dist_matrix_plot, aes(x=Var2, y=Var1, fill=value)) + 
  geom_tile() + 
  viridis::scale_fill_viridis(limits=c(0,1), na.value=&quot;white&quot;) + 
  theme(
    axis.text.x = element_text(hjust = 0.95, angle = 20),
    panel.background = element_rect(fill = &quot;white&quot;, color = &quot;white&quot;)
  ) + 
  labs(x = &quot;&quot;, y = &quot;&quot;, fill = &quot;BC-Distance&quot;) +
  geom_text(aes(Var2, Var1, label = round(value,2)), color = &quot;white&quot;, size = 3)</code></pre>
<p>Assessing the ARGs found in the pangenome of the mock-community, DARTE-QM showed high precision with the mean distance between all observed-samples found to be <strong>0.1</strong>, whereas the accuracy compared to our theoretical distribution was weaker, with a mean distance of <strong>0.52</strong>.</p>
<p><img src="mock_analysis_files/figure-html/distances_out-1.png" width="1152" /> <br> <br></p>
</div>
<div id="synthetic-oligonucleotide-spike" class="section level2 tabset tabset-pills">
<h2>Synthetic Oligonucleotide ‘Spike’</h2>
<pre class="r"><code>graph_data &lt;- melt(spike_counts, id.vars = c(&#39;ARG_Family&#39;,&#39;ARG&#39;, &#39;Primer&#39;))
setnames(graph_data, colnames(graph_data[,-c(1:3)]), c(&quot;Sample&quot;, &quot;Spike&quot;))
graph_data[, Spike_Level := as.numeric(gsub(&#39;-&#39;,&#39;.&#39;,
              gsub(&#39;Mock-&#39;,&#39;&#39;, gsub(&#39;spike-.?&#39;,&#39;&#39;, metadata$ID[match(as.character(graph_data$Sample), metadata$Sample)]))))]
setkey(graph_data, &#39;Spike_Level&#39;)
graph_data[, Color := schuylR::create_palette(length(unique(graph_data[[&#39;Spike_Level&#39;]])))[as.factor(graph_data[[&#39;Spike_Level&#39;]])]]</code></pre>
<p>The added levels of the spike-sequence were detected with a linear relation to their laboratory concentrations with an <span class="math inline">\(R^{2}\)</span> = 0.91.</p>
<div id="linear-regression" class="section level3">
<h3>Linear Regression</h3>
<pre class="r"><code>spike_points &lt;-  ggplot(graph_data, aes_string(x = &#39;Spike_Level&#39;, y = &#39;Spike&#39;)) + 
  geom_smooth(formula = y~x, method = &quot;lm&quot;, se = FALSE, aes(x = Spike_Level, y = Spike), color = &#39;black&#39; ) + 
  geom_point(stat = &quot;identity&quot;, size = 4, alpha = 1, position = position_dodge(width = 0), color = graph_data$Color) + 
  theme_bw() + 
  theme(axis.text.x = element_text(hjust = 1, size = 10), 
    axis.text.y = element_text(hjust = 0.95, size = 10), 
    axis.title.x = element_text(size = 12, face = &quot;bold&quot;), 
    axis.title.y = element_text(size = 12, face = &quot;bold&quot;), 
    axis.ticks.x = element_blank(), 
    legend.title = element_text(size = 10, face = &quot;bold&quot;), 
    legend.text = element_text(size = 8), 
    legend.spacing.x = unit(0.005, &quot;npc&quot;), 
    legend.key.size = unit(4, &quot;mm&quot;), 
    legend.background = element_rect(fill = (alpha = 0)), 
    panel.background = element_rect(color = &quot;black&quot;, size = 1.5, fill = &quot;white&quot;), 
    panel.spacing = unit(0.015, &quot;npc&quot;), 
    strip.text.x = element_text(size = 10, face = &quot;bold&quot;, color = &quot;black&quot;), 
    strip.background = element_rect(colour = &quot;black&quot;, size = 1.4, fill = &quot;white&quot;)) + 
  labs(x = &quot;Spike Level&quot;, y = &quot;Spike Sequences Detected at &gt;= 98% Identity&quot;)</code></pre>
<p><img src="mock_analysis_files/figure-html/spike_counts_line_reg-1.png" width="576" /> <br> <br></p>
</div>
<div id="bar-chart" class="section level3">
<h3>Bar Chart</h3>
<pre class="r"><code>spike_bars &lt;-  ggplot(graph_data,
                      aes_string(x = &#39;Sample&#39;, y = &#39;Spike&#39;)) + 
  geom_bar(stat = &quot;identity&quot;, size = 0.45, alpha = 1, position = &quot;dodge&quot;,
           fill = graph_data$Color) + 
  theme_light() +
  theme(
    axis.line.x = element_line(
      colour = &#39;black&#39;,
      size = 1,
      linetype = &#39;solid&#39;
    ),
    axis.line.y = element_line(
      colour = &#39;black&#39;,
      size = 1,
      linetype = &#39;solid&#39;
    ),
    axis.text.x = element_text(
      size = 10,
      vjust = 1,
      hjust = 1,
      angle = 30
    ),
    axis.title.x = element_text(size = 12, face = &quot;bold&quot;),
    axis.title.y = element_text(size = 12, face = &quot;bold&quot;),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12, face = &quot;bold&quot;),
    legend.background = element_rect(fill = (alpha = 0)),
    legend.key.size = unit(4, &quot;mm&quot;),
    legend.spacing.x = unit(0.005, &#39;npc&#39;),
    strip.text.x = element_text(size = 10, face = &#39;bold&#39;, color = &#39;black&#39;),
    strip.background = element_rect(colour = &#39;black&#39;, size = 1.4, fill = &#39;white&#39;)
  ) +
  scale_y_continuous(expand = expansion(mult = c(0.0025, 0.01))) +
  ylab(&quot;Read Count of Spike Sequence&quot;) + 
  xlab(&quot;&quot;)</code></pre>
<p><img src="mock_analysis_files/figure-html/spike_counts_bars-1.png" width="576" /> <br> <br></p>
</div>
<div id="table" class="section level3">
<h3>Table</h3>
<div id="htmlwidget-2c331d61784e94a0b92b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2c331d61784e94a0b92b">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike"],["spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike"],["spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike","spike"],["Mock, 2","Mock, 3","Mock + spike (0.0025), 1","Mock + spike (0.0025), 2","Mock + spike (0.0025), 3","Mock + spike (0.009), 1","Mock + spike (0.009), 2","Mock + spike (0.009), 3","Mock + spike (0.025), 1","Mock + spike (0.025), 2","Mock + spike (0.12), 1","Mock + spike (0.12), 2","Mock + spike (0.12), 3","Mock + spike (0.25), 1","Mock + spike (0.25), 2","Mock + spike (0.25), 3"],[0,0,21,11,12,37,71,88,221,150,631,562,939,2016,1677,1047],[0,0,0.0025,0.0025,0.0025,0.009,0.009,0.009,0.025,0.025,0.12,0.12,0.12,0.25,0.25,0.25]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ARG_Family<\/th>\n      <th>ARG<\/th>\n      <th>Primer<\/th>\n      <th>Sample<\/th>\n      <th>Spike<\/th>\n      <th>Spike_Level<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br> <br></p>
</div>
</div>
<div id="unexpected-args" class="section level2 tabset tabset-pills">
<h2>Unexpected ARGs</h2>
<div id="args" class="section level3">
<h3>ARGs</h3>
<pre class="r"><code>classification = &quot;ARG&quot;
graph_data &lt;- mock_observed_counts[!(get(classification) %in% mock_expected_counts[[classification]])]

graph_data &lt;- melt(graph_data, id.vars = colnames(graph_data)[1:3], variable.name = &#39;Sample&#39;)
graph_data &lt;- graph_data[, sum(value), by = c(&#39;Sample&#39;, classification)]
set(graph_data, j = &#39;Sample&#39;, value = factor(graph_data[[&#39;Sample&#39;]], levels = mock_samples))

graph_data[[classification]] &lt;- factor(graph_data[[classification]], 
                                       levels = c(&#39;spike&#39;, rev(unique(graph_data[[classification]][!(graph_data[[classification]]==&#39;spike&#39;)]))))
set(graph_data, which(graph_data[[&#39;V1&#39;]] &lt; 1), j=&#39;V1&#39;, value=NA)

unexpected_graph &lt;- ggplot(graph_data, aes_string(&quot;Sample&quot;, classification, fill = &quot;V1&quot;)) + 
  geom_tile(color = &quot;white&quot;, size = 0.25) +
  guides(colour = guide_legend(ncol = ceiling(length(
    unique(graph_data[[classification]])
  ) / 25))) +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size = 10), 
        axis.text.y = element_text(hjust = 0.95, size = 10), 
        axis.title.x = element_text(size = 10, face = &quot;bold&quot;), 
        axis.title.y = element_text(size = 10, face = &quot;bold&quot;), 
        axis.ticks.x = element_blank(), legend.title = element_text(size = 8, face = &quot;bold&quot;), 
        legend.text = element_text(size = 8), 
        legend.spacing.x = unit(0.005, &quot;npc&quot;), 
        legend.key.size = unit(6, &quot;mm&quot;), 
        legend.background = element_rect(fill = (alpha = 0), color = &quot;black&quot;, size = 0.25), 
        panel.background = element_rect(color = &quot;black&quot;, size = 1.4), 
        strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
        strip.background = element_rect(colour = &quot;black&quot;, size = 1.4)) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.53)) + 
  labs(x = &quot;Sample&quot;, fill = &quot;Abundance&quot;) + 
  ggraph::scale_fill_viridis(limits = c(0.0001, max(graph_data$V1, na.rm = T))) </code></pre>
<p><img src="mock_analysis_files/figure-html/unexpected_ARG_graph_data_out-1.png" width="1152" /></p>
</div>
<div id="primers" class="section level3">
<h3>Primers</h3>
<p>DARTE-QM also detected 65 primers that were not found in the sequenced genomes of the mock-community.</p>
<pre class="r"><code>classification = &quot;Primer&quot;
graph_data &lt;- mock_observed_counts[Primer %in% unexpected_primers]
graph_data &lt;- melt(graph_data, id.vars = colnames(graph_data)[1:3], variable.name = &#39;Sample&#39;)
graph_data &lt;- graph_data[, sum(value), by = c(&#39;Sample&#39;, classification)]
set(graph_data, j = &#39;Sample&#39;, value = factor(graph_data[[&#39;Sample&#39;]], levels = mock_samples))

graph_data[[classification]] &lt;- factor(graph_data[[classification]], 
                                       levels = c(&#39;spike&#39;, rev(unique(graph_data[[classification]][!(graph_data[[classification]]==&#39;spike&#39;)]))))
set(graph_data, which(graph_data[[&#39;V1&#39;]] &lt; 1), j=&#39;V1&#39;, value=NA)

unexpected_graph &lt;- ggplot(graph_data, aes_string(&quot;Sample&quot;, classification, fill = &quot;V1&quot;)) + 
  geom_tile(color = &quot;white&quot;, size = 0.25) +
  guides(colour = guide_legend(ncol = ceiling(length(
    unique(graph_data[[classification]])
  ) / 25))) +
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 30, hjust = 1, size = 10), 
        axis.text.y = element_text(hjust = 0.95, size = 10), 
        axis.title.x = element_text(size = 10, face = &quot;bold&quot;), 
        axis.title.y = element_text(size = 10, face = &quot;bold&quot;), 
        axis.ticks.x = element_blank(), 
        legend.title = element_text(size = 8, face = &quot;bold&quot;), 
        legend.text = element_text(size = 8), 
        legend.spacing.x = unit(0.005, &quot;npc&quot;), 
        legend.key.size = unit(6, &quot;mm&quot;), 
        legend.background = element_rect(fill = (alpha = 0), color = &quot;black&quot;, size = 0.25), 
        panel.background = element_rect(color = &quot;black&quot;, size = 1.4), 
        strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
        strip.background = element_rect(colour = &quot;black&quot;, size = 1.4)) + 
  scale_x_discrete(expand = expansion(mult = 0, add = 0.53)) + 
  labs(x = &quot;Sample&quot;, fill = &quot;Abundance&quot;) + 
  ggraph::scale_fill_viridis(limits = c(0.0001, max(graph_data$V1, na.rm = T))) </code></pre>
<p><img src="mock_analysis_files/figure-html/unexpected_graph_data_out-1.png" width="1152" /></p>
</div>
</div>

<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
